---
title: "publication_code"
author: "ben_young"
date: "03/12/2019"
output: html_document
---

# Library Loading

```{r package installation and loading, include=FALSE}
library(stringr)
library(dplyr)
library(tibble)
library(tidyr)
library(tximport)
library(tximportData)
library(readr)
library(DESeq2)
library(pheatmap)
library(RColorBrewer)
library(vsn)
library(genefilter)
library(gplots)
library(ggplot2)
library(WGCNA)
library(limma)
library(flashClust)
library(pathfindR)
library(edgeR)
library(ComplexHeatmap)
library(tidyr)
library(data.table)
library(GOplot)
library(clusterProfiler)
library(enrichplot)
library(RSvgDevice)
library(dendextend)
library(ape)
library(circlize)
library(CEMiTool)
library(extrafont)
```

```{r font importing, include=F}
#font_import()
fonts()
loadfonts()
```


# Functions for Analysis
## Modified PCA Functions from DeSeq2  

```{r Function for PC 2 and 3, include=FALSE}
#PCA 2 and 3 axis creation
pcaaxes23 = function (object, intgroup = "condition", ntop = 500, returnData = FALSE) 
{
    rv <- rowVars(assay(object))
    select <- order(rv, decreasing = TRUE)[seq_len(min(ntop, 
        length(rv)))]
    pca <- prcomp(t(assay(object)[select, ]))
    percentVar <- pca$sdev^2/sum(pca$sdev^2)
    if (!all(intgroup %in% names(colData(object)))) {
        stop("the argument 'intgroup' should specify columns of colData(dds)")
    }
    intgroup.df <- as.data.frame(colData(object)[, intgroup, 
        drop = FALSE])
    group <- if (length(intgroup) > 1) {
        factor(apply(intgroup.df, 1, paste, collapse = ":"))
    }
    else {
        colData(object)[[intgroup]]
    }
    d <- data.frame(PC2 = pca$x[, 2], PC3 = pca$x[, 3], group = group, 
        intgroup.df, name = colnames(object))
    if (returnData) {
        attr(d, "percentVar") <- percentVar[2:3]
        return(d)
    }
    ggplot(data = d, aes_string(x = "PC2", y = "PC3", color = "group")) + 
        geom_point(size = 3) + xlab(paste0("PC2: ", round(percentVar[2] * 
        100), "% variance")) + ylab(paste0("PC3: ", round(percentVar[3] * 
        100), "% variance")) + coord_fixed()
}
```

```{r Function for PC 3 and 4, include=FALSE}
#PCA 3 and 4 axis creation
pcaaxes34 = function (object, intgroup = "condition", ntop = 500, returnData = FALSE) 
{
    rv <- rowVars(assay(object))
    select <- order(rv, decreasing = TRUE)[seq_len(min(ntop, 
        length(rv)))]
    pca <- prcomp(t(assay(object)[select, ]))
    percentVar <- pca$sdev^2/sum(pca$sdev^2)
    if (!all(intgroup %in% names(colData(object)))) {
        stop("the argument 'intgroup' should specify columns of colData(dds)")
    }
    intgroup.df <- as.data.frame(colData(object)[, intgroup, 
        drop = FALSE])
    group <- if (length(intgroup) > 1) {
        factor(apply(intgroup.df, 1, paste, collapse = ":"))
    }
    else {
        colData(object)[[intgroup]]
    }
    d <- data.frame(PC3 = pca$x[, 3], PC4 = pca$x[, 4], group = group, 
        intgroup.df, name = colnames(object))
    if (returnData) {
        attr(d, "percentVar") <- percentVar[3:4]
        return(d)
    }
    ggplot(data = d, aes_string(x = "PC3", y = "PC4", color = "group")) + 
        geom_point(size = 3) + xlab(paste0("PC3: ", round(percentVar[3] * 
        100), "% variance")) + ylab(paste0("PC4: ", round(percentVar[4] * 
        100), "% variance")) + coord_fixed()
}

```


## Bingo Heatmap Formatting Function  

```{r Function for Formatting Bingo Heatmaps}
pad27 <- function(x){
  str_x = as.character(x)
  digits = nchar(str_x)
  toAdd = 7-digits
  name = paste0('GO:',
                strrep(0,toAdd), 
                x)
  name
}
```


## Cytoscape Edge and Node Export Function

```{r expoeritng all modules edge and node for cytoscape, eval= TRUE}
exportAll2Cytoscape <- function(cols, TOM, data, minweight = -1, maxedges = 30000){
  modules = cols
  modules = unique(modules)
  modules[modules != "grey"]
  
  
  
  lapply(modules, function(x){
    probes = names(data)
    inModule = is.finite(match(moduleColors, x))
    modProbes = probes[inModule]    
    #modProbes = extract_tx(modProbes)
    modGenes = annot$Gene.Annotation[match(modProbes, annot$Gene.ID)]
    # Select the corresponding Topological Overlap
    modTOM = TOM[inModule, inModule]
    dimnames(modTOM) = list(modProbes, modProbes)
    nGenes = length(modProbes)
    edgeprob = (maxedges/(nGenes*nGenes))
    if(edgeprob >1){
      edgeprob = 1
    }
    if(edgeprob <0){
      edgeprob = 1
    }
    
    if(minweight <0 | minweight*(nGenes*nGenes) > maxedges){
      weight = quantile(modTOM,
                            probs = 1- edgeprob,
                            names = FALSE)
    }
    else{
      weight = quantile(modTOM,
                            probs = minweight,
                            names = FALSE)
    }
    print(c(edgeprob, weight, maxedges/edgeprob ))
    # Export the network into edge and node list files Cytoscape can read
    WGCNA::exportNetworkToCytoscape(modTOM,
                                    edgeFile = paste(paste0("apal_WGCNA_cytoscapeinput_edge"), paste(x), ".txt", sep=""),
                                    nodeFile = paste(paste0("apal_WGCNA_cytoscapeinput_node"), paste(x), ".txt", sep=""),
                                    weighted = TRUE,
                                    threshold = weight,
                                    nodeNames = modProbes,
                                    altNodeNames = modGenes,
                                    nodeAttr = moduleColors[inModule])
  })
  
}
```


# Data Import and Pre-analysis Filtering

```{r GFF Annotation File Import}
Annotation_file_for_analysis <- read.delim("~/Dropbox/stuff/Annotation_file_for_analysis.txt")
annot <- Annotation_file_for_analysis
str(annot)
annot %>%
  mutate(Gene.ID = toupper(Gene.ID)) -> annot
```

```{r Treatment File Import, include=FALSE}
#reading in the treatment file 
#NB: treatment file needs to be same order as the counts in txi.salmon.countall$counts
tfall <- read.csv("~/Dropbox/stuff/all_treatment_file.csv")
str(tfall)
tfall$Genotype = as.factor(tfall$Genotype)
tfall$Treatment_basic = as.factor(tfall$Treatment_basic)
tfall$Year = as.factor(tfall$Year)
str(tfall)
#View(tfall)
attach(tfall)
```

```{r custom annotation file for bingo GO analysis}
#annot %>% dplyr::select(Gene.ID, GO_Terms_Swiss.Prot) %>% 
#  tidyr::drop_na() %>%
#  tidyr::separate_rows(GO_Terms_Swiss.Prot) %>%
#  filter(!str_detect(GO_Terms_Swiss.Prot, 'GO')) %>%
#  unite(x, c(Gene.ID, GO_Terms_Swiss.Prot), sep = " = ", remove = T) -> 
#  caf_GO

#write.table(caf_GO,
#            file = "~/Dropbox/PhD/Projects/Acropora_palmata_disease_exposure/paper/GO_gene_lists/CAF_GO.txt",
#            quote=F, 
#            row.names=F, 
#            col.names=F)

## add in this line as line 1 in text file manually
## 
```

```{r Making Salmon Vectors Names, include=FALSE}
#Loading quantification data output from the slippery Salmon
setwd("~/Dropbox/stuff/host_quant/")
PPall <- list.files(path = "~/Dropbox/stuff/host_quant//", full.names = F, pattern = "\\_salmon$")

FILESall <- file.path(PPall, "quant.sf")

names(FILESall) <- PPall
head(FILESall)
all(file.exists(FILESall))
```

```{r Reading in Salmon Files, include=F}
#Importing the count data using salmon quant.sf files and the text to gene file
setwd("~/Dropbox/stuff/host_quant//")
txi.salmon.countall <- tximport(FILESall, type = "salmon", txOut = TRUE )
str(txi.salmon.countall$counts)
#View(txi.salmon.countall$counts)
```

```{r Making DeSeq object, include=FALSE}
#making the dds model to use in deseq2
ddsall = DESeqDataSetFromTximport(txi.salmon.countall, tfall, ~ Treatment_basic)
```

```{r CPM filtering and DDS object creation}
ddsall = DESeqDataSetFromTximport(txi.salmon.countall, tfall, ~ Treatment_basic)

nrow(ddsall)
ncol(ddsall)

# cpm filtering step and seeing what original VS filtered gene number left is
cccall <- counts(ddsall)
keep <- rowSums(cpm(cccall)>=1) >= 15
cccall <- cccall[keep, ]

nrow(ddsall)
ncol(ddsall)
nrow(cccall)
ncol(cccall)

cccall %>%
  as.data.frame() %>%
  rownames_to_column(var="Gene.ID") %>%
  mutate(Gene.ID = toupper(Gene.ID)) %>%
  column_to_rownames(var = "Gene.ID") -> cccall
  
ddsall <- DESeqDataSetFromMatrix(countData = cccall, colData = tfall, design = ~ Treatment_basic)

ddsall$group1 <- factor(paste0(ddsall$Treatment_basic, ddsall$Outcome))
design(ddsall) <- ~ Year + group1
```


# Gene Expression Analysis
## Variance stabilised transformations
```{r VST, include=FALSE}
## VSD stabilisation and visulisation
vsdall <- vst(ddsall, blind=FALSE)
```

```{r VST with limma removing year, include = F}
vsdyear <- vst(ddsall)
assay(vsdyear) <- limma::removeBatchEffect(assay(vsdyear), vsdyear$Year)
```

## Principal Component Analysis
```{r PCA Analysis 1 and 2, echo = F}
cbPalette <- c("dodgerblue3" ,"darkgoldenrod2", "orangered3")
plotPCA(vsdall, intgroup=c("Treatment"), returnData = F)

pca12 <- plotPCA(vsdall, intgroup=c("Treatment"), returnData = TRUE)
ggplot(pca12, aes(PC1, PC2, color=Treatment, shape=Year)) + 
  geom_point(size=3) +  xlab(paste0("PC1 45% variance")) + 
  ylab(paste0("PC2 13% variance")) + 
  theme(text = element_text(size=11, family = "Arial"), 
        legend.position = "right", 
        panel.background = element_rect(fill = "transparent"), 
        axis.line = element_line(colour = "black"), 
        legend.key = element_rect(fill = "transparent"), 
        axis.text = element_text(size = 8))  + 
  theme(legend.key.size = unit(0.7, "cm")) + 
  scale_colour_manual(values=cbPalette) +
  stat_ellipse(aes(PC1, PC2, group=Year), type = "norm") +
  scale_color_manual(name = "Treatments",
                     values = c("dodgerblue3" ,"darkgoldenrod2", "orangered3"),
                     labels=c("Baseline", "No Transmission", "Transmission"))

#ggsave(filename = "pc12_treat_3.eps", 
#       path = "~/Dropbox/Projects/Acropora_palmata_disease_exposure/paper/figures/",
#       width = 10)
```

```{r PCA Analysis 2 and 3, echo = F}
cbPalette <- c("dodgerblue3" ,"darkgoldenrod2", "orangered3")
pcaaxes23(vsdall, intgroup=c("Treatment"), returnData = F)
pca23 <- pcaaxes23(vsdall, intgroup=c("Treatment"), returnData = TRUE)
ggplot(pca23, aes(PC2, PC3, colour=Treatment)) + 
  geom_point(size=3) +  
  xlab(paste0("PC2 13% variance")) + 
  ylab(paste0("PC3 8% variance")) + 
  theme(text = element_text(size=11, family = "Arial"), 
        legend.position = "right", 
        panel.background = element_rect(fill = "transparent"), 
        axis.line = element_line(colour = "black"), 
        legend.key = element_rect(fill = "transparent"), 
        axis.text = element_text(size = 8))  +  
  scale_colour_manual(values=cbPalette) +
  stat_ellipse() +
  scale_color_manual(name = "Treatments",
                     values = c("dodgerblue3" ,"darkgoldenrod2", "orangered3"),
                     labels=c("Baseline", "No Transmission", "Transmission"))
#ggsave(filename = "pc23_treat_3.jpeg", path = "~/Dropbox/Projects/Acropora_palmata_disease_exposure/figures/")
```

```{r PCA Analysis 3 and 4, echo = F}
pca34 <- pcaaxes34(vsdall, intgroup=c("Treatment_basic"), returnData = TRUE)
ggplot(pca34, aes(PC3, PC4, colour=Treatment_basic)) + 
  geom_point(size=4) +  
  xlab(paste0("PC3 8% variance")) + 
  ylab(paste0("PC4 4% variance")) + 
  theme(legend.position="right")  + 
  theme(text = element_text(size=10))  + 
  theme(legend.key.size = unit(0.5, "cm")) + 
  geom_point(size = 4) +
  scale_colour_manual(values=cbPalette) +
  theme_bw() +
  stat_ellipse()

pca34 <- pcaaxes34(vsdall, intgroup=c("Treatment_basic"), returnData = TRUE)
ggplot(pca34, aes(PC3, PC4, colour=Genotype, shape=Treatment_basic)) + 
  geom_point(size=3) +  
  xlab(paste0("PC3 8% variance")) + 
  ylab(paste0("PC4 4% variance")) + 
  theme(legend.position="right")  + 
  theme(text = element_text(size=6))  + 
  theme(legend.key.size = unit(0.1, "cm")) + 
  geom_point(size = 3) +
  theme_classic() + 
  geom_polygon(fill=NA)
```


## Genes Driving PC1 Separation
```{r Genes driving difference between PC1 and PC2} 
## getting the laodings for the PC's from PCA plots
rv <- rowVars(assay(vsdall))
#View(rowvars)
ntop = 1000
select <- order(rv, decreasing = TRUE)[seq_len(min(ntop, 
        length(rv)))]
#View(select)

intgroup=c("Year", "Treatment_basic")
ntop = 1000
rv <- rowVars(assay(vsdall))
    select <- order(rv, decreasing = TRUE)[seq_len(min(ntop,
        length(rv)))]
    pca <- prcomp(t(assay(vsdall)[select, ]))
    percentVar <- pca$sdev^2/sum(pca$sdev^2)
    if (!all(intgroup %in% names(colData(vsdall)))) {
        stop("the argument 'intgroup' should specify columns of colData(dds)")
    }
    
rotations <- pca$rotation
pc1_rotations <- rotations[,c(1)]
pc1_rotations <- as.data.frame(pc1_rotations)
str(pc1_rotations)

#interquartile range and selecting any genes outside of it to explain pc1 high variance explanation
quantile(pc1_rotations$pc1_rotations, 0.90)
quantile(pc1_rotations$pc1_rotations, 0.10)
quantile(pc1_rotations$pc1_rotations, 0.50)
median(pc1_rotations$pc1_rotations)

##quantiles for most variable genes
wanted_values <- subset(pc1_rotations, pc1_rotations >0.02404031  | pc1_rotations < -0.04734249 )
nrow(pc1_rotations)
nrow(wanted_values)

rownamespc1_variance <- row.names(wanted_values)
#write.table(
#  rownamespc1_variance,
#  file = "~/Dropbox/Projects/Acropora_palmata_disease_exposure/paper/GO_gene_lists/PC1_genes4GO.txt",
#  sep = "\t",
#  quote = F,
#  row.names = F,
#  col.names = F
#)
```

```{r Annotated PC1 genes, include = F}
annot %>% dplyr::select(Gene.ID, Gene.Annotation) %>% 
  inner_join(wanted_values %>% 
               as.data.frame() %>% 
               rownames_to_column(var="Gene.ID")) ->
  PC1_annotate
#write.csv(PC1_annotated, 
#          file = "~/Dropbox/Projects/Acropora_palmata_disease_exposure/paper/annotated_genelists/PC1_annotated.csv",
#          row.names = F)
```

NB. Run Bingo analysis in cytoscape here.

```{r Importing Results file from Bingo, include = F}
read.table(
  file = "/Users/benyoung/Dropbox/PhD/Projects/apal_innate_immune_response/host_analysis/GO_gene_lists/BINGO_results/PC1_host_allGO_FDR_0.01.bgo",
  skip = 19, #Need to modify this value on depending on each Bingo result table
  stringsAsFactors = F,
  sep = "\t",
  header = T
) %>%
  dplyr::select(GO.ID, Description, Genes.in.test.set) %>%
  separate_rows(Genes.in.test.set, sep = '\\|', convert = T) -> PC1_results
```

```{r Genes from identified GO Analysis of PC1, include = F}
wantGO <- c("GO:0070062", "GO:0005840", "GO:0006412", "GO:0042254", "GO:0006364", "GO:0003735", "GO:0034185", "GO:0016337")
#cell cell adhesion, translation, extracellular matrix iorganisaton, viral reporudctive process. 

#Putting the GO and numbers before terms for imported bingo results
PC1_results %>% 
  mutate(GO.ID_FULL = pad27(GO.ID)) %>%
  dplyr::filter(GO.ID_FULL %in% wantGO) ->
  intGOterms

forheatmap <- as.character(intGOterms$Genes.in.test.set)

annot %>%
  dplyr::select(Gene.ID, Gene.Annotation) %>%
  inner_join(intGOterms %>% dplyr::rename(Gene.ID = Genes.in.test.set)) -> PC1_GO_gene

# write.csv(PC1_GO_gene, file = "~/Dropbox/Projects/Acropora_palmata_disease_exposure/paper/annotated_genelists/PC1_host_annotated_GOlist.csv")
```

```{r PC1 variance complexheatmap prep, include = F}
matmatnorder <- assay(vsdall)[forheatmap,]
View(forheatmap)
View(assay(vsdall))

# Colours for the years
ccann <- data.frame(tfall$Year)
colnames(ccann) <- c("Year")
colcol <- list("Year" = c("2016" = "grey", "2017" = "black"))

# Top annotation bar for the complex heatmap
samname <- HeatmapAnnotation(df = ccann,
                             which = "col",
                             col = colcol,                        
                             simple_anno_size = unit(1.5, "cm"),
                             annotation_name_gp = gpar(fontsize = 70),
                             annotation_name_side = "left")

# Fill for the gene VST data
col_fun = colorRamp2(c(-2, 0, 2), 
                     c("dodgerblue4", "white", "red3"))
col_fun(seq(-3, 3))

# adding annotation into vsd count matrix
annot %>% 
  dplyr::select(Gene.ID, Gene.Annotation) %>% 
  right_join(matmatnorder %>% 
               as.data.frame() %>% 
               rownames_to_column(var="Gene.ID")) %>%
  column_to_rownames(var="Gene.ID") %>%
  tidyr::drop_na() -> matmatnorder

# GO matrix (yes no) for genes matching to a term
PC1_GO_gene %>%
  dplyr::select(Gene.Annotation, Description, Gene.ID) %>%
  dplyr::mutate(membership=1) %>%
  tidyr::spread(Description, membership, fill=0) -> PC1_GO2gene_matrix

# merging by Gene.ID to make 1 matrix for heatmaps
matmatnorder %>%
  rownames_to_column(var="Gene.ID") %>%
  inner_join(PC1_GO2gene_matrix) %>%
  column_to_rownames(var="Gene.ID") -> matmatnorder

# Changing names to a character
matmatnorder$Gene.Annotation <- as.character(matmatnorder$Gene.Annotation)

#id_to_annot, need to have rownames when putting into complex heatmap
hello <- rowAnnotation(foo = anno_text(matmatnorder %>% as.data.frame() %>% dplyr::select(Gene.Annotation)))
```

```{r PC1 Variance Heatmaps, fig.height=50, fig.width=40, echo = F}
# NB manual specify of the GO columns, here it is 86:93. First heatmap we remove, 2nd heatmap we select. 
HM1 <-
  Heatmap(
    t(scale(
      t(matmatnorder %>% as.data.frame() %>% dplyr::select(-c(
        "Gene.Annotation", 86:93
      )))
    )),
    cluster_columns = T,
    show_row_dend = F,
    show_column_dend = T,
    column_km = 2,
    row_km = 2,
    col = col_fun,
    row_title = NULL,
    column_title = NULL,
    top_annotation = samname,
    column_gap = unit(2, "cm"),
    row_gap = unit(2, "cm"),
    column_dend_height = unit(20, "cm"),
    row_dend_width = unit(8, "cm"),
    show_row_names = F,
    heatmap_width = unit(70, "cm"),
    heatmap_height = unit(200, "cm"),
    row_title_gp = gpar(fontsize = 80),
    column_title_gp = gpar(fontsize = 80),
    show_column_names = F
  )

#GO yes no heatmap
HM2 <-
  Heatmap(
    t(t(
      matmatnorder %>% as.data.frame() %>% dplyr::select(c(86:93))
    )),
    show_row_dend = F,
    show_column_dend = F,
    heatmap_width = unit(95, "cm"),
    show_column_names = T,
    show_row_names = F,
    right_annotation = rowAnnotation(foo = anno_text(
      matmatnorder$Gene.Annotation,  gp = gpar(fontsize = 70, font = 10)
    )),
    column_names_gp = gpar(fontsize = 50, font = 1),
    col = c("white", "black"),
    column_names_rot = 30
  )

ht_list <- HM1 + HM2

draw(ht_list, ht_gap = unit(1, "cm"))


```


## Analysis of gene expression due to visual health status
### DeSeq2 Analysis of Health Status

```{r Differential Exprssion analysis, echo = F}
ddsall <- DESeq(ddsall)
resall <- results(ddsall, alpha = 0.01)
summary(resall)

resultsNames(ddsall)
ddsall$group1

## between treatments
btall <- results(ddsall, 
                 contrast = c("group1", "exposedtransmission", "baselinebaseline"), 
                 alpha = 0.05, 
                 test = "Wald")

bntall <- results(ddsall, 
                  contrast = c("group1", "exposedno_transmission", "baselinebaseline"), 
                  alpha = 0.05, 
                  test = "Wald")

summary(btall)
summary(bntall)
```

```{r LFC ><1 and p < 0.01 for BvsNT and BvsT, include = F}
## selecting padg<0.01
sigbtall <- subset(btall, padj<0.01)
sortedbtall <- sigbtall[order(sigbtall$padj), ]
allrowsbt <- row.names(sortedbtall) 
sortedbtall <- as.data.frame(sortedbtall)

sigbntall <- subset(bntall, padj<0.01)
sortedbntall <- sigbntall[order(sigbntall$padj), ]
allrowsbnt <- row.names(sortedbntall) 
sortedbntall <- as.data.frame(sortedbntall)

## subsetting for L2FC < -1 and > 1
sortedbtLFC1 <- subset(sortedbtall, log2FoldChange >= 1 | log2FoldChange <= -1)
sortedbntLFC1 <- subset(sortedbntall, log2FoldChange >= 1 | log2FoldChange <= -1)

## making rows for GO analysis
allrowsbt <- row.names(sortedbtLFC1)
allrowsbnt <- row.names(sortedbntLFC1) 

#write.table(allrowsbnt, 
#            sep = "\t", 
#            file = "~/Dropbox/Projects/Acropora_palmata_disease_exposure/paper/GO_gene_lists/BNT_all4go.txt",
#            quote=F, 
#            row.names=F, 
#            col.names=F)

#write.table(allrowsbt, 
#            sep = "\t", 
#            file = "~/Dropbox/Projects/Acropora_palmata_disease_exposure/paper/GO_gene_lists/BT_all4go.txt",
#            quote=F, 
#            row.names=F, 
#            col.names=F)
```

```{r Venn of BvsNT and BvsT, echo = F}
venn(list(allrowsbt, allrowsbnt))
```

```{r Unique gene sets for BvsNT and BvsT, include = F}
## shared genes
treat_shared <- intersect(allrowsbnt, allrowsbt)
rownames(treat_shared) <- colnames(treat_shared)

## unique b vs nt
str(sortedbtLFC1)
rownames_to_column(sortedbtLFC1, var = "geneid") %>%
  filter(!geneid %in% treat_shared) %>%
  column_to_rownames(var="geneid") -> 
  unique_bt_lfc1
btlfc1_unique <- rownames(unique_bt_lfc1)
rownames(btlfc1_unique) <- colnames(btlfc1_unique)
#View(btlfc1_unique)

## unique b vs nt
str(sortedbntLFC1)
rownames_to_column(sortedbntLFC1, var = "geneid") %>%
  filter(!geneid %in% treat_shared) %>%
  column_to_rownames(var="geneid") -> 
  unique_bnt_lfc1
bntlfc1_unique <- rownames(unique_bnt_lfc1)
rownames(bntlfc1_unique) <- colnames(bntlfc1_unique)

#write.table(treat_shared, 
#            sep = "\t", 
#            file = "~/Dropbox/Projects/Acropora_palmata_disease_exposure/paper/GO_gene_lists/shared_DEG4GO.txt",
#            quote=F, 
#            row.names=F, 
#            col.names=F)

#write.table(btlfc1_unique, 
#            file = "~/Dropbox/Projects/Acropora_palmata_disease_exposure/paper/GO_gene_lists/bt_gene4GO.txt",
#            sep = "\t", 
#            quote=F, 
#            row.names=F, 
#            col.names=F)

#write.table(bntlfc1_unique, 
#            file = "~/Dropbox/Projects/Acropora_palmata_disease_exposure/paper/GO_gene_lists/bnt_gene4GO.txt",
#            sep = "\t", 
#            quote=F, 
#            row.names=F, 
#            col.names=F)
```

```{r Annotated Gene Lists for unique, include = F}
#Baseline Transmission
annot %>% dplyr::select(Gene.ID, Gene.Annotation) %>% 
  inner_join(unique_bt_lfc1 %>% 
               rownames_to_column(var="Gene.ID") %>%
               as.data.frame()) -> BT_annotated

##Baseline No Transmission
annot %>% dplyr::select(Gene.ID, Gene.Annotation) %>% 
  inner_join(unique_bnt_lfc1 %>% 
               rownames_to_column(var="Gene.ID") %>%
               as.data.frame()) -> BNT_annotated

#write.csv(BT_annotated, 
#          file = "~/Dropbox/Projects/Acropora_palmata_disease_exposure/paper/annotated_genelists/BT_DEG0.01_LFC1_annotated.csv",
#          row.names = F)


#write.csv(BNT_annotated, 
#          file = "~/Dropbox/Projects/Acropora_palmata_disease_exposure/paper/annotated_genelists/BNT_DEG0.01_LFC1_annotated.csv",
#          row.names = F)
```

```{r Common genes from venn annotated with LFC directionality, include = F}
rownames_to_column(sortedbntLFC1, var = "geneid") %>%
  filter(geneid %in% treat_shared) %>%
  column_to_rownames(var="geneid") -> bnt_shared
bnt_shared$DEG <- "bnt"

annot %>% dplyr::select(Gene.ID, Gene.Annotation) %>% 
  right_join(bnt_shared %>% 
               as.data.frame() %>% 
               rownames_to_column(var="Gene.ID")) -> bnt_shared_annot
bnt_shared_annot$Gene.ID <- as.factor(bnt_shared_annot$Gene.ID)

rownames_to_column(sortedbtLFC1, var = "geneid") %>%
  filter(geneid %in% treat_shared) %>%
  column_to_rownames(var="geneid") -> bt_shared
bt_shared$DEG <- "bt"

annot %>% dplyr::select(Gene.ID, Gene.Annotation) %>% 
  right_join(bt_shared %>% 
               as.data.frame() %>% 
               rownames_to_column(var="Gene.ID")) -> bt_shared_annot
bt_shared_annot$Gene.ID <- as.factor(bt_shared_annot$Gene.ID)

shared <- rbind(bnt_shared_annot, bt_shared_annot)
#View(shared)
#write.csv(shared, file = "~/Desktop/shared.csv")
```


### GO analysis DeSeq2 Health Status Results 
#### Baseline Vs No Transmission

```{r Importing Bingo results file, include=F}
#allrowsbnt
read.table(
  file = "/Users/benyoung/Dropbox/PhD/Projects/apal_innate_immune_response/host_analysis/GO_gene_lists/BINGO_results/allBNT_allGO_FDR_0.01.bgo",
  skip = 23, # change value depending on each bingo results file
  stringsAsFactors = F,
  sep = "\t",
  header = T
) %>%
  dplyr::select(GO.ID, Description, Genes.in.test.set) %>%
  separate_rows(Genes.in.test.set, sep = '\\|', convert = T) -> bnt_results
```

```{r GO term selection and prep for visualisation, include = F}
# Put the GO terms you are interested in here from Bingo results to visualise in heatmap as well as get genes in terms
wantGO <- c("GO:0007155", "GO:0007166", "GO:0044420")

#Putting the GO and numbers before terms for imported bingo results
bnt_results %>% 
  mutate(GO.ID_FULL = pad27(GO.ID)) %>%
  dplyr::filter(GO.ID_FULL %in% wantGO) ->
  intGOterms
#View(intGOterms)

# Genes for the heat map
forheatmap <- as.character(intGOterms$Genes.in.test.set)

# Annotating the genes in the GO terms to see genes
annot %>%
  dplyr::select(Gene.ID, Gene.Annotation) %>%
  inner_join(intGOterms %>% dplyr::rename(Gene.ID = Genes.in.test.set)) -> PC1_GO_gene
```

```{r BNT Genes in GO terms Heatmap Prep}
mat_bnt <- assay(vsdyear)[forheatmap,]

tfall %>% 
  dplyr::select(Treatment) %>% 
  dplyr::filter(Treatment != c("exposed.transmission")) %>%
  as.data.frame() -> bnt_tfall

ccann <- bnt_tfall
colnames(ccann) <- c("Treatment")
colcol <- list("Treatment"=c("baseline"="dodgerblue3", "exposed.no_transmission"="darkgoldenrod2"))
samname <- HeatmapAnnotation(df = ccann,
                             which = "col",
                             col = colcol,
                             simple_anno_size = unit(1.5, "cm"),
                             annotation_name_gp = gpar(fontsize = 70),
                             annotation_name_side = "left")

col_fun = colorRamp2(c(-2, 0, 2), c("dodgerblue4", "white", "red3"))
col_fun(seq(-3, 3))

# adding annotation into vsd count matrix
#t(scale(t(mat_bnt)))
annot %>% dplyr::select(Gene.ID, Gene.Annotation) %>% 
  right_join(mat_bnt %>% 
               as.data.frame() %>% 
               rownames_to_column(var="Gene.ID")) %>%
  column_to_rownames(var="Gene.ID") %>%
  tidyr::drop_na() -> matmatnorder-> mat_bnt_order

# GO matrix (yes no) for genes matching to a term
PC1_GO_gene %>%
  dplyr::select(Gene.Annotation, Description, Gene.ID) %>%
  dplyr::mutate(membership=1) %>%
  tidyr::spread(Description, membership, fill=0) -> PC1_GO2gene_matrix

# merging by Gene.ID to make 1 matrix for heatmaps
mat_bnt_order %>%
  rownames_to_column(var="Gene.ID") %>%
  inner_join(PC1_GO2gene_matrix) %>%
  column_to_rownames(var="Gene.ID") -> mat_bnt_order
```

```{r BNT Heatmap Visualisation, fig.width=60, fig.height=55, echo = F}
HM1 <-
  Heatmap(
    t(scale(
      t(mat_bnt_order %>% as.data.frame() %>% dplyr::select(
        -c(
          "Gene.Annotation",
          "11_salmon",
          "12_salmon",
          "13_salmon",
          "17_salmon",
          "34_salmon",
          "35_salmon",
          "39_salmon",
          "41_salmon",
          "43_salmon",
          "45_salmon",
          "47_salmon",
          "49_salmon",
          "51_salmon",
          "53_salmon",
          "55_salmon",
          "57_salmon",
          "59_salmon",
          "61_salmon",
          "63_salmon",
          "65_salmon",
          "82_salmon",
          "83_salmon",
          "87_salmon",
          "88_salmon",
          "91_salmon",
          "92_salmon",
          "93_salmon",
          86:88
        )
      ))
    )),
    show_row_dend = F,
    show_column_dend = T,
    cluster_columns = T,
    col = col_fun,
    cluster_rows = T,
    top_annotation = samname,
    column_km = 3,
    row_km = 2,
    column_gap = unit(2, "cm"),
    row_gap = unit(2, "cm"),
    column_dend_height = unit(20, "cm"),
    row_dend_width = unit(8, "cm"),
    show_row_names = F,
    heatmap_width = unit(100, "cm"),
    heatmap_height = unit(250, "cm"),
    row_title_gp = gpar(fontsize = 80),
    column_title_gp = gpar(fontsize = 80),
    show_column_names = F,
    row_title = NULL, 
    column_title = NULL
  )

HM2 <-
  Heatmap(
    t(t(
      mat_bnt_order %>% as.data.frame() %>% dplyr::select(c(86:88))
    )),
    show_row_dend = F,
    show_column_dend = F,
    heatmap_width = unit(93, "cm"),
    show_column_names = T,
    show_row_names = F,
    right_annotation = rowAnnotation(foo = anno_text(
      mat_bnt_order$Gene.Annotation,
      gp = gpar(fontsize = 60, font = 10)
    )),
    column_names_gp = gpar(fontsize = 50, font = 1),
    col = c("white", "black"),
    column_names_rot = 20
  )

ht_list <- HM1 + HM2
draw(ht_list, ht_gap = unit(1, "cm"))
```

### Baseline Vs Transmission

```{r Importing Bingo results for BvsT, include = F}
read.table(
  file = "/Users/benyoung/Dropbox/PhD/Projects/apal_innate_immune_response/host_analysis/GO_gene_lists/BINGO_results/allBT_allGO_FDR_0.01.bgo",
  skip = 26, # change depending on your Bingo results
  stringsAsFactors = F,
  sep = "\t",
  header = T
) %>%
  dplyr::select(GO.ID, Description, Genes.in.test.set) %>%
  separate_rows(Genes.in.test.set, sep = '\\|', convert = T) -> bt_results
```

```{r GO term select and prep for visualisation}
# wantGO <- c("GO:0008218", "GO:0042744", "GO:0008543", "GO:0007178", "GO:0042035", "GO:0008201", "GO:0005125", "GO:0016209", "GO:0006952")
# wantGO <- c("GO:0004930","GO:0008227", "GO:0008528", "GO:0008188")
wantGO <- c("GO:0006952", "GO:0005125", "GO:0042744")

#Putting the GO and numbers before terms for imported bingo results
bt_results %>% 
  mutate(GO.ID_FULL = pad27(GO.ID)) %>%
  dplyr::filter(GO.ID_FULL %in% wantGO) ->
  intGOterms

forheatmap <- as.character(intGOterms$Genes.in.test.set)

annot %>%
  dplyr::select(Gene.ID, Gene.Annotation) %>%
  inner_join(intGOterms %>% dplyr::rename(Gene.ID = Genes.in.test.set)) -> PC1_GO_gene
```

```{r BvsT Heatmap Prep, include = F}
mat_bnt <- assay(vsdyear)[forheatmap,]

tfall %>% 
  dplyr::select(Treatment) %>% 
  dplyr::filter(Treatment != c("exposed.no_transmission")) %>%
  as.data.frame() -> bnt_tfall

ccann <- bnt_tfall
colnames(ccann) <- c("Treatment")
colcol <- list("Treatment"=c("baseline"="dodgerblue3", "exposed.transmission"="orangered3"))
samname <- HeatmapAnnotation(df = ccann,
                             which = "col",
                             col = colcol,
                             simple_anno_size = unit(1.5, "cm"),
                             annotation_name_gp = gpar(fontsize = 70),
                             annotation_name_side = "left")


col_fun = colorRamp2(c(-2, 0, 2), c("dodgerblue4", "white", "red3"))
col_fun(seq(-3, 3))

# adding annotation into vsd count matrix
#t(scale(t(mat_bnt)))
annot %>% dplyr::select(Gene.ID, Gene.Annotation) %>% 
  right_join(mat_bnt %>% 
               as.data.frame() %>% 
               rownames_to_column(var="Gene.ID")) %>%
  column_to_rownames(var="Gene.ID") %>%
  tidyr::drop_na() -> matmatnorder-> mat_bnt_order

# GO matrix (yes no) for genes matching to a term
PC1_GO_gene %>%
  dplyr::select(Gene.Annotation, Description, Gene.ID) %>%
  dplyr::mutate(membership=1) %>%
  tidyr::spread(Description, membership, fill=0) -> PC1_GO2gene_matrix

# merging by Gene.ID to make 1 matrix for heatmaps
mat_bnt_order %>%
  rownames_to_column(var="Gene.ID") %>%
  inner_join(PC1_GO2gene_matrix) %>%
  column_to_rownames(var="Gene.ID") -> mat_bnt_order
```

```{r BvsT heatmap visualisation, fig.width=40, fig.height=50, echo = F}
HM1 <- Heatmap(t(scale(t(mat_bnt_order %>% as.data.frame() %>% dplyr::select(-c("Gene.Annotation", "1_salmon", "14_salmon", "18_salmon", "19_salmon", "2_salmon", "20_salmon", "23_salmon", "24_salmon", "25_salmon", "28_salmon", "29_salmon", "3_salmon", "30_salmon", "37_salmon", "67_salmon", "7_salmon", "8_salmon", "84_salmon", "9_salmon", 86:88))))),
        show_row_dend = F,
        show_column_dend = T,
        cluster_columns = T,
        col = col_fun,
        cluster_rows = T,
        top_annotation = samname,
        column_km = 2,
        row_km = 2,
        column_gap = unit(2, "cm"),
        row_gap = unit(2, "cm"),
        column_dend_height = unit(20, "cm"),
        row_dend_width = unit(3, "cm"),
        show_row_names = F,
        heatmap_width = unit(70, "cm"),
        heatmap_height = unit(200, "cm"),    
        row_title_gp = gpar(fontsize = 80),
        column_title_gp = gpar(fontsize = 80),
        show_column_names = F, 
        row_title = NULL,
        column_title = NULL)

HM2 <-
  Heatmap(
    t(t(
      mat_bnt_order %>% as.data.frame() %>% dplyr::select(c(86:88))
    )),
    show_row_dend = F,
    show_column_dend = F,
    heatmap_width = unit(75, "cm"),
    show_column_names = T,
    show_row_names = F,
    right_annotation = rowAnnotation(foo = anno_text(
      mat_bnt_order$Gene.Annotation,
      gp = gpar(fontsize = 70, font = 10)
    )),
    column_names_gp = gpar(fontsize = 50, font = 1),
    col = c("white", "black"),
    column_names_rot = 20
  )

ht_list <- HM1 + HM2
draw(ht_list, ht_gap = unit(1, "cm"))
```


## WGCNA analysis without year
This is from the tutorial I avaliable at https://horvath.genetics.ucla.edu/html/CoexpressionNetwork/Rpackages/WGCNA/Tutorials/

### Running WGCNA Analysis
```{r SUPER DUPER IMPORTANT FOR WGCNA TO DO THIS, include = F}
options(stringsAsFactors = FALSE)
```

```{r Genes for WGCNA and checking they all a-okay, echo = F}
genes4wgcna <- assay(vsdyear)
genes4wgcna <- t(genes4wgcna)
genes4wgcna <- as.data.frame(genes4wgcna)

dim(genes4wgcna)

gsg = goodSamplesGenes(genes4wgcna, verbose = 3)
gsg$allOK
```

```{r Treatment File Object for WGCNA, echo = F}
#4. Create an object with the treatment file
#Create an object called "datTraits" that contains your trait data
allTraits = read.csv("~/Dropbox/PhD/Projects/apal_innate_immune_response/host_analysis/WGCNA_analysis/wgcna_treat_file.csv")
head(allTraits)
str(allTraits)
#form a data frame analogous to expression data that will hold the clinical traits.
rownames(allTraits) = allTraits$Sample_numbers
allTraits$Sample_numbers = NULL
table(rownames(allTraits)==rownames(genes4wgcna)) #should return TRUE if datasets align correctly, otherwise your names are out of order
```

```{r Checking for outliers, fig.height=20, fig.width=20, echo = F}
sampleTree = hclust(dist(genes4wgcna), method = "ward");
# Plot the sample tree: Open a graphic output window of size 12 by 9 inches
# The user should change the dimensions if the window is too large or too small.
sizeGrWindow(12,9)
#pdf(file = "Plots/sampleClustering.pdf", width = 12, height = 9);
par(cex = 1);
par(mar = c(0,4,2,0))
plot(sampleTree, main = "Sample clustering to detect outliers", sub="", xlab="", cex.lab = 1.5,
cex.axis = 1.5, cex.main = 2)
# Plot a line to show the cut
abline(h = 300, col = "red");
# Determine cluster under the line
clust = cutreeStatic(sampleTree, cutHeight = 500, minSize = 10)
table(clust)
# clust 1 contains the samples we want to keep.
keepSamples = (clust==1)
datExpr = genes4wgcna[keepSamples, ]
nGenes = ncol(datExpr)
nSamples = nrow(datExpr)
```

```{r Identyfying Soft Power, echo = F}
# Choose a set of soft-thresholding powers
powers = c(c(1:10), seq(from = 12, to=20, by=2))
# Call the network topology analysis function
sft = pickSoftThreshold(datExpr, powerVector = powers, verbose = 5)
# Plot the results:
sizeGrWindow(9, 5)
par(mfrow = c(1,2));
cex1 = 0.9;
# Scale-free topology fit index as a function of the soft-thresholding power
plot(sft$fitIndices[,1], -sign(sft$fitIndices[,3])*sft$fitIndices[,2],
xlab="Soft Threshold (power)",ylab="Scale Free Topology Model Fit,signed R^2",type="n",
main = paste("Scale independence"));
text(sft$fitIndices[,1], -sign(sft$fitIndices[,3])*sft$fitIndices[,2],
labels=powers,cex=cex1,col="red");
# this line corresponds to using an R^2 cut-off of h
abline(h=0.90,col="red")
# Mean connectivity as a function of the soft-thresholding power
plot(sft$fitIndices[,1], sft$fitIndices[,5],
xlab="Soft Threshold (power)",ylab="Mean Connectivity", type="n",
main = paste("Mean connectivity"))
text(sft$fitIndices[,1], sft$fitIndices[,5], labels=powers, cex=cex1,col="red")
```

```{r Signed adjacency matrix using soft power from last step, echo = F}
softPower = 12;
adjacency = adjacency(datExpr, type="signed", power = softPower);
```

```{r Generation of topological overlap, echo = F}
# Turn adjacency into topological overlap
TOM = TOMsimilarity(adjacency);
dissTOM = 1-TOM
```

```{r Gene clustering with TOM-based dissimilarity plot, fig.width=15, fig.height=15, echo = F}
# Call the hierarchical clustering function
geneTree = hclust(as.dist(dissTOM), method = "average");
# Plot the resulting clustering tree (dendrogram)
sizeGrWindow(12,9)
plot(geneTree, xlab="", sub="", main = "Gene clustering on TOM-based dissimilarity",
labels = FALSE, hang = 0.04);
```

```{r Module identification, echo = F}
# We like large modules, so we set the minimum module size relatively high:
minModuleSize = 60;
# Module identification using dynamic tree cut:
dynamicMods = cutreeDynamic(dendro = geneTree, distM = dissTOM,
deepSplit = 2, pamRespectsDendro = FALSE,
minClusterSize = minModuleSize);
table(dynamicMods)
```

```{r Gene dendogram and module colours plot, echo = F}
# Convert numeric lables into colors
dynamicColors = labels2colors(dynamicMods)
table(dynamicColors)
# Plot the dendrogram and colors underneath
sizeGrWindow(8,6)
plotDendroAndColors(geneTree, dynamicColors, "Dynamic Tree Cut",
dendroLabels = FALSE, hang = 0.03,
addGuide = TRUE, guideHang = 0.05,
main = "Gene dendrogram and module colors")
```

```{r Merging similar modules at 0.40, fig.height=10, fig.width=15, echo = F}
# Calculate eigengenes
MEList = moduleEigengenes(datExpr, colors = dynamicColors)
MEs = MEList$eigengenes
# Calculate dissimilarity of module eigengenes
MEDiss = 1-cor(MEs);
# Cluster module eigengenes
METree = hclust(as.dist(MEDiss), method = "average");
# Plot the result
sizeGrWindow(7, 6)

plot(METree, 
     main = "Clustering of module eigengenes",
     xlab = "", 
     sub = "")
MEDissThres = 0.40
# Plot the cut line into the dendrogram
abline(h=MEDissThres, col = "red")
# Call an automatic merging function
merge = mergeCloseModules(datExpr, dynamicColors, cutHeight = MEDissThres, verbose = 3)
# The merged module colors
mergedColors = merge$colors;
# Eigengenes of the new merged modules:
mergedMEs = merge$newMEs;
```

```{r Fig 5a Cluster dendogram with original and merged modules, fig.height=3, fig.width=10, echo = F}
sizeGrWindow(12, 5)
plotDendroAndColors(geneTree, cbind(dynamicColors, mergedColors),
c("Dynamic Tree Cut", "Merged dynamic"),
dendroLabels = FALSE, hang = 0.03,
addGuide = TRUE, guideHang = 0.05)
```

```{r Eigengenes for each sample to each module, echo = F}
# Rename to moduleColors
moduleColors = mergedColors
# Construct numerical labels corresponding to the colors
colorOrder = c("grey", standardColors(50));
moduleLabels = match(moduleColors, colorOrder)-1;
MEs = mergedMEs;
MEs
```

```{r Making Variables for Plotting, echo = F}
# Define numbers of genes and samples
nGenes = ncol(datExpr);
nSamples = nrow(datExpr);
# Recalculate MEs with color labels
datTraits <- allTraits
MEs0 = moduleEigengenes(datExpr, moduleColors)$eigengenes
MEs = orderMEs(MEs0)
moduleTraitCor = cor(MEs, datTraits, use = "p");
moduleTraitPvalue = corPvalueStudent(moduleTraitCor, nSamples);
```

### WGCNA Visualisation of Results
```{r All modules module-trait relationship plot, fig.height=15, fig.width=15, echo = F}
sizeGrWindow(10,6)
# Will display correlations and their p-values
textMatrix = paste(signif(moduleTraitCor, 2), "\n(",
signif(moduleTraitPvalue, 1), ")", sep = "");
dim(textMatrix) = dim(moduleTraitCor)
par(mar = c(6, 8.5, 3, 3));
# Display the correlation values within a heatmap plot
#View(moduleTraitCor)
#View(textMatrix)

labeledHeatmap(Matrix = moduleTraitCor,
  xLabels = names(datTraits),
  yLabels = names(MEs),
  ySymbols = names(MEs),
  colorLabels = FALSE,
  colors = blueWhiteRed(50),
  textMatrix = textMatrix,
  setStdMargins = FALSE,
  cex.text = 1,
  zlim = c(-1,1),
  main = paste("Module-trait relationships"))
```

```{r Each module hub genes, echo = F}
tophub <- chooseTopHubInEachModule(genes4wgcna, moduleColors, omitColors = "grey", power = 10)

tophub <- as.data.frame(tophub)
tophub$Gene.ID <- tophub$tophub

annot %>% 
  dplyr::select(Gene.ID, Gene.Annotation) %>% 
  right_join(tophub %>% 
               as.data.frame() %>%
               rownames_to_column(var="module")) %>%
  column_to_rownames(var="tophub") -> 
               tophub_annotated
             
# View(tophub_annotated)
# write.csv(tophub_annotated, 
#           file = "~/Dropbox/Projects/Acropora_palmata_disease_exposure/paper/annotated_genelists/hub_genes_annotated.csv", 
#           row.names = F)
```

```{r Figure 5b heatmap, fig.width=8, echo = F}
str(moduleTraitCor)
moduleTraitCor %>%
  as.data.frame() %>%
  rownames_to_column('rowname') %>%
  dplyr::slice(4,13,17, 1, 2, 3, 11, 12) %>%
  column_to_rownames('rowname') ->
  int_modules

int_modules[4,3] = 0
int_modules[5,3] = 0
int_modules[6,3] = 0
int_modules[7,2] = 0
int_modules[8,2] = 0

textMatrix %>%
  as.data.frame() %>%
  dplyr::slice(4,13,17, 1, 2, 3, 11, 12) ->
  int_modules_text

int_modules_text[4,3] = 0
int_modules_text[5,3] = 0
int_modules_text[6,3] = 0
int_modules_text[7,2] = 0
int_modules_text[8,2] = 0

MEs %>%
  dplyr::select(c("MElightgreen", "MEbrown", "MEskyblue", "MEcyan", "MEgrey60", "MEmediumpurple3", "MEblack", "MEdarkolivegreen")) -> ME_just3

datTraits
#View(int_modules)
#View(int_modules_text)

par(mar = c(6, 8.5, 3, 3));
labeledHeatmap(Matrix = int_modules,
  xLabels = c("Baseline", "Exposed: No Transmission", "Exposed: Transmission"),
  yLabels = names(ME_just3),
  ySymbols = names(ME_just3),
  colorLabels = FALSE,
  colors = blueWhiteRed(50),
  textMatrix = int_modules_text,
  setStdMargins = FALSE,
  cex.text = 1.5,
  cex.lab.y = 0.6,
  cex.lab.x = 0.7,
  zlim = c(-1,1),
  main = paste("Module-trait relationships"),
  las = 0)
```

```{r Fig 5b gene count bargraph, fig.height=10, fig.width=4}
# nrow(brown)
# nrow(lightgreen)
# nrow(skyblue)
# nrow(black)
# nrow(mediumpurple3)
# nrow(grey60)
# nrow(cyan)
# nrow(darkolivegreen)

gene_bg <- cbind(c(366, 1656, 220, 838, 1003, 97, 1423, 183))
colnames(gene_bg) <- c("genes")
rownames(gene_bg) <- c("Lightgreen", "Brown", "Skyblue", "Cyan", "Grey60", "Mediumpurple", "Black", "Darkolivegreen")
#View(gene_bg)
gene_bg %>% 
  as.data.frame() %>% 
  rownames_to_column(var = "module") -> gene_bg
gene_bg$module <- factor(gene_bg$module, c("Lightgreen", "Brown", "Skyblue", "Cyan", "Grey60", "Mediumpurple", "Black", "Darkolivegreen"))
str(gene_bg)

ggplot(data = gene_bg, (aes(module, genes, fill = module))) +
  geom_bar(stat = "identity", 
           fill = c("lightgreen", "brown", "skyblue", "cyan", "grey60", "purple", "black", "darkolivegreen"),
           position = "dodge") +
  theme_classic() +
  theme(strip.text.x = element_blank(), 
        panel.border=element_blank(),
        axis.line.x = element_blank(),
        axis.line=element_line(),
        axis.title.y = element_blank(),
        axis.text.y.left = element_blank(),
        axis.line.y = element_blank(),
        axis.ticks.y = element_blank(),
        legend.position = "none") +
  scale_y_continuous(position = "left",
                     limits = c(0, 1800), breaks = seq(0, 1800, by = 300)) +
  coord_flip()
```

```{r Figure 5c, echo = F}
# Define variable weight containing the weight column of datTrait
weight = as.data.frame(datTraits$exposed.no_transmission);
names(weight) = "weight"
# names (colors) of the modules
modNames = substring(names(MEs), 3)
geneModuleMembership = as.data.frame(cor(datExpr, MEs, use = "p"));
MMPvalue = as.data.frame(corPvalueStudent(as.matrix(geneModuleMembership), nSamples));
names(geneModuleMembership) = paste("MM", modNames, sep="");
names(MMPvalue) = paste("p.MM", modNames, sep="");
geneTraitSignificance = as.data.frame(cor(datExpr, weight, use = "p"));
GSPvalue = as.data.frame(corPvalueStudent(as.matrix(geneTraitSignificance), nSamples));
names(geneTraitSignificance) = paste("GS.", names(weight), sep="");
names(GSPvalue) = paste("p.GS.", names(weight), sep="");

geneTraitSignificance %>%
  rownames_to_column(var="Gene.ID") %>%
  full_join(geneModuleMembership %>%
              rownames_to_column(var = "Gene.ID") %>%
              dplyr::select(MMbrown, MMlightgreen, MMskyblue, MMcyan, MMgrey60, MMmediumpurple3, Gene.ID)) %>%
  tidyr::gather(module, module_membership, MMbrown:MMmediumpurple3) %>%
  ggplot(aes(module_membership, GS.weight, color = module)) +
  geom_point(size = 2, alpha = 1/100) +
  geom_smooth(color="black", method = lm, se = T, fill = "grey") +
  theme_classic() +
  ylim(c(-1,1)) +
  xlim(c(-1,1)) +
  scale_color_manual(values=c("brown", "cyan", "lightgrey", "lightgreen", "purple", "skyblue")) +
  facet_wrap(~module, scales = "free") +
  theme(strip.text.x = element_blank()) +
  theme(panel.border=element_blank(), axis.line=element_line(), legend.position = "none") +
  labs(x="Module Membership",
       y="Gene Significance for Exposed: No Transmission")
```

```{r Figure 5d, echo = F}
# Define variable weight containing the weight column of datTrait
weight = as.data.frame(datTraits$exposed.transmission);
names(weight) = "weight"
# names (colors) of the modules
modNames = substring(names(MEs), 3)
geneModuleMembership = as.data.frame(cor(datExpr, MEs, use = "p"));
MMPvalue = as.data.frame(corPvalueStudent(as.matrix(geneModuleMembership), nSamples));
names(geneModuleMembership) = paste("MM", modNames, sep="");
names(MMPvalue) = paste("p.MM", modNames, sep="");
geneTraitSignificance = as.data.frame(cor(datExpr, weight, use = "p"));
GSPvalue = as.data.frame(corPvalueStudent(as.matrix(geneTraitSignificance), nSamples));
names(geneTraitSignificance) = paste("GS.", names(weight), sep="");
names(GSPvalue) = paste("p.GS.", names(weight), sep="");

geneTraitSignificance %>%
  rownames_to_column(var="Gene.ID") %>%
  full_join(geneModuleMembership %>%
              rownames_to_column(var = "Gene.ID") %>%
              dplyr::select(MMbrown, MMlightgreen, MMskyblue, MMdarkolivegreen, MMblack, Gene.ID)) %>%
  tidyr::gather(module, module_membership, MMbrown:MMblack) %>%
  ggplot(aes(module_membership, GS.weight, color = module)) +
  geom_point(size = 2, alpha = 1/100) +
  geom_smooth(color="black", method = lm) +
  theme_classic() +
  ylim(c(-1,1)) +
  xlim(c(-1,1)) +
  scale_color_manual(values=c("grey22", "brown", "darkgreen", "lightgreen", "skyblue")) +
  facet_wrap(~module, scales = "free") +
  theme(strip.text.x = element_blank()) +
  theme(panel.border=element_blank(), axis.line=element_line(), legend.position = "none") +
  labs(x="Module Membership",
       y="Gene Significance for Exposed: Transmission")
```

```{r Genes and annotations for significant WGCNA modules T and NT, include = F}
## SKyblue module
skyblue <- names(genes4wgcna)[moduleColors == "skyblue"]
#write.table(
#  skyblue,
#  file = "~/Dropbox/Projects/Acropora_palmata_disease_exposure/paper/GO_gene_lists/WGCNA/skyblue.txt",
#  quote = F,
#  col.names = F,
#  row.names = F
#)

skyblue <- as.data.frame(skyblue)
skyblue$Gene.ID <- skyblue$skyblue

annot %>%
  dplyr::select(Gene.ID, Gene.Annotation) %>%
  right_join(skyblue) %>%
  column_to_rownames(var = "skyblue") ->
  skyblue_annotated
#View(skyblue_annotated)
#write.csv(skyblue_annotated, 
#          file = "~/Dropbox/Projects/Acropora_palmata_disease_exposure/paper/annotated_genelists/skyblue_module_annotated.csv", 
#          row.names = F)

## Lightgreen module
lightgreen <- names(genes4wgcna)[moduleColors == "lightgreen"]
#write.table(
#  lightgreen,
#  file = "~/Dropbox/Projects/Acropora_palmata_disease_exposure/paper/GO_gene_lists/WGCNA/lightgreen.txt",
#  quote = F,
#  col.names = F,
#  row.names = F
#)

lightgreen <- as.data.frame(lightgreen)
lightgreen$Gene.ID <- lightgreen$lightgreen

annot %>%
  dplyr::select(Gene.ID, Gene.Annotation) %>%
  right_join(lightgreen) %>%
  column_to_rownames(var = "lightgreen") ->
  lightgreen_annotated
#View(lightgreen_annotated)
#write.csv(lightgreen_annotated, 
#          file = "~/Dropbox/Projects/Acropora_palmata_disease_exposure/paper/annotated_genelists/lightgreen_module_annotated.csv", 
#          row.names = F)


## Brown Module
brown <- names(genes4wgcna)[moduleColors == "brown"]
#write.table(
#  brown,
#  file = "~/Dropbox/Projects/Acropora_palmata_disease_exposure/paper/GO_gene_lists/WGCNA/brown.txt",
#  quote = F,
#  col.names = F,
#  row.names = F
#)

brown <- as.data.frame(brown)
brown$Gene.ID <- brown$brown

annot %>%
  dplyr::select(Gene.ID, Gene.Annotation) %>%
  right_join(brown) %>%
  column_to_rownames(var = "brown") ->
  brown_annotated
#View(brown_annotated)
#write.csv(brown_annotated, 
#          file = "~/Dropbox/Projects/Acropora_palmata_disease_exposure/paper/annotated_genelists/brown_module_annotated.csv", 
#          row.names = F)

nrow(brown)
```

```{r Genes and annotations for significant WGCNA modules Transmission, include = F}
## Black Module
black <- names(genes4wgcna)[moduleColors == "black"]
#write.table(
#  black,
#  file = "~/Dropbox/Projects/Acropora_palmata_disease_exposure/paper/GO_gene_lists/WGCNA/black.txt",
#  quote = F,
#  col.names = F,
#  row.names = F
#)

black <- as.data.frame(black)
black$Gene.ID <- black$black

annot %>%
  dplyr::select(Gene.ID, Gene.Annotation) %>%
  right_join(black) %>%
  column_to_rownames(var = "black") ->
  black_annotated
#View(black_annotated)
#write.csv(black_annotated, 
#          file = "~/Dropbox/Projects/Acropora_palmata_disease_exposure/paper/annotated_genelists/black_module_annotated.csv", 
#          row.names = F)


## Grey60 module
grey60 <- names(genes4wgcna)[moduleColors == "grey60"]
#write.table(
#  grey60,
#  file = "~/Dropbox/Projects/Acropora_palmata_disease_exposure/paper/GO_gene_lists/WGCNA/grey60.txt",
#  quote = F,
#  col.names = F,
#  row.names = F
#)

grey60 <- as.data.frame(grey60)
grey60$Gene.ID <- grey60$grey60

annot %>%
  dplyr::select(Gene.ID, Gene.Annotation) %>%
  right_join(grey60) %>%
  column_to_rownames(var = "grey60") ->
  grey60_annotated
#View(grey60_annotated)
#write.csv(grey60_annotated, 
#          file = "~/Dropbox/Projects/Acropora_palmata_disease_exposure/paper/annotated_genelists/grey60_module_annotated.csv", 
#          row.names = F)

## Darkolivegreen module
darkolivegreen <- names(genes4wgcna)[moduleColors == "darkolivegreen"]
#write.table(
#  darkolivegreen,
#  file = "~/Dropbox/Projects/Acropora_palmata_disease_exposure/paper/GO_gene_lists/WGCNA/darkolivegreen.txt",
#  quote = F,
#  col.names = F,
#  row.names = F
#)

darkolivegreen <- as.data.frame(darkolivegreen)
darkolivegreen$Gene.ID <- darkolivegreen$darkolivegreen

annot %>%
  dplyr::select(Gene.ID, Gene.Annotation) %>%
  right_join(darkolivegreen) %>%
  column_to_rownames(var = "darkolivegreen") ->
  darkolivegreen_annotated
#View(darkolivegreen_annotated)
#write.csv(darkolivegreen_annotated, 
#          file = "~/Dropbox/Projects/Acropora_palmata_disease_exposure/paper/annotated_genelists/darkolivegreen_module_annotated.csv", 
#          row.names = F)

nrow(grey60)
nrow(black)
nrow(darkolivegreen)
```

```{r Genes and annotations for significant WGCNA modules NT, include = F}
## Lightcyan Module
lightcyan1 <- names(genes4wgcna)[moduleColors == "lightcyan1"]
#write.table(
#  lightcyan1,
#  file = "~/Dropbox/Projects/Acropora_palmata_disease_exposure/paper/GO_gene_lists/WGCNA/lightcyan1.txt",
#  quote = F,
#  col.names = F,
#  row.names = F
#)

lightcyan1 <- as.data.frame(lightcyan1)
lightcyan1$Gene.ID <- lightcyan1$lightcyan1

annot %>%
  dplyr::select(Gene.ID, Gene.Annotation) %>%
  right_join(lightcyan1) %>%
  column_to_rownames(var = "lightcyan1") ->
  lightcyan1_annotated
#View(lightcyan1_annotated)
#write.csv(lightcyan1_annotated, 
#          file = "~/Dropbox/Projects/Acropora_palmata_disease_exposure/paper/annotated_genelists/lightcyan1_module_annotated.csv", 
#          row.names = F)


## Cyan Module
cyan <- names(genes4wgcna)[moduleColors == "cyan"]
#write.table(
#  cyan,
#  file = "~/Dropbox/Projects/Acropora_palmata_disease_exposure/paper/GO_gene_lists/WGCNA/cyan.txt",
#  quote = F,
#  col.names = F,
#  row.names = F
#)

cyan <- as.data.frame(cyan)
cyan$Gene.ID <- cyan$cyan

annot %>%
  dplyr::select(Gene.ID, Gene.Annotation) %>%
  right_join(cyan) %>%
  column_to_rownames(var = "cyan") ->
  cyan_annotated
#View(cyan_annotated)
#write.csv(cyan_annotated, 
#          file = "~/Dropbox/Projects/Acropora_palmata_disease_exposure/paper/annotated_genelists/cyan_module_annotated.csv", 
#          row.names = F)


#Mediumpurple3 module
mediumpurple3 <- names(genes4wgcna)[moduleColors == "mediumpurple3"]
#write.table(
#  mediumpurple3,
#  file = "~/Dropbox/Projects/Acropora_palmata_disease_exposure/paper/GO_gene_lists/WGCNA/mediumpurple3.txt",
#  quote = F,
#  col.names = F,
#  row.names = F
#)

mediumpurple3 <- as.data.frame(mediumpurple3)
mediumpurple3$Gene.ID <- mediumpurple3$mediumpurple3

annot %>%
  dplyr::select(Gene.ID, Gene.Annotation) %>%
  right_join(mediumpurple3) %>%
  column_to_rownames(var = "mediumpurple3") ->
  mediumpurple3_annotated
#View(mediumpurple3_annotated)
#write.csv(mediumpurple3_annotated, 
#          file = "~/Dropbox/Projects/Acropora_palmata_disease_exposure/paper/annotated_genelists/mediumpurple3_module_annotated.csv", 
#          row.names = F)
```


### Bingo Analysis for WGCNA modules
```{r Bingo results formatting for heatmaps BROWN MODULE, include = F}
read.table(
  file = "/Users/benyoung/Dropbox/Projects/Acropora_palmata_disease_exposure/paper/GO_gene_lists/BINGO_results/brown_allGO_FDR_0.01.bgo",
  skip = 27,
  stringsAsFactors = F,
  sep = "\t",
  header = T
) %>%
  dplyr::select(GO.ID, Description, Genes.in.test.set) %>%
  separate_rows(Genes.in.test.set, sep = '\\|', convert = T) -> brown_results



#wantGO <- c("GO:0035355", "GO:0002102", "GO:0045335")
#wantGO <- c("GO:0005539", "GO:0071723")
#wantGO <- c("GO:0009617", "GO:0042035", "GO:0050778", "GO:0042088", "GO:0034150", "GO:0002755")
#wantGO <- c("GO:0048584")
#wantGO <- c("GO:0007186")
#wantGO <- c("GO:0030139")
wantGO <- c("GO:0032493", "GO:0014070", "GO:0009593", "GO:0042088", "GO:0042035", "GO:0002221", "GO:0050778", "GO:0002253", "GO:0035355",
            "GO:0007186", "GO:0005539", "GO:0004930", "GO:0016176", "GO:0071723", "GO:0030139", "GO:0045335", "GO:0030666", "GO:0030670",
            "GO:0002102", "GO:0006800")

#Putting the GO and numbers before terms for imported bingo results
brown_results %>% 
  mutate(GO.ID_FULL = pad27(GO.ID)) %>%
  dplyr::filter(GO.ID_FULL %in% wantGO) ->
  intGOterms
View(intGOterms)

forheatmap <- as.character(intGOterms$Genes.in.test.set)


colnames(brown_results)[colnames(brown_results)=="Genes.in.test.set"] <- "Gene.ID"
annot %>%
  dplyr::select(Gene.ID, Gene.Annotation) %>%
  inner_join(intGOterms %>% dplyr::rename(Gene.ID =Genes.in.test.set)) %>% View()
    #write.csv(., file = "~/Desktop/brown_GO_genes.csv")

#View(brown_results)
#View(forheatmap)


# Putting Gene.ID to upper case so can match to bingo results
#annot %>%
#  mutate(GENE.ID = toupper(Gene.ID)) %>%
#  dplyr::select(GENE.ID, GO_Terms_Swiss.Prot) %>%
#    tidyr::drop_na() %>% View()
```

```{r Brown Module GO genes, fig.width=15, fig.height=15, echo = F}
mat_bnt <- assay(vsdyear)[forheatmap,]
#View(mat_bnt)

tfall %>% 
  dplyr::select(Treatment) %>% 
  as.data.frame() -> bnt_tfall

ccann <- bnt_tfall
colnames(ccann) <- c("Treatment")
colcol <- list("Treatment"=c("baseline"="dodgerblue3", "exposed.no_transmission"="darkgoldenrod2", "exposed.transmission"="orangered3"))
samname <- HeatmapAnnotation(df = ccann,
                             which = "col",
                             col = colcol)

col_fun = colorRamp2(c(-2, 0, 2), c("dodgerblue4", "white", "red3"))
col_fun(seq(-3, 3))

t(scale(t(mat_bnt)))
annot %>% dplyr::select(Gene.ID, Gene.Annotation) %>% 
  right_join(mat_bnt %>% 
               as.data.frame() %>% 
               rownames_to_column(var="Gene.ID")) %>%
  column_to_rownames(var="Gene.ID") %>%
  tidyr::drop_na() -> matmatnorder -> mat_bnt_order


Heatmap(t(scale(t(mat_bnt_order %>% as.data.frame() %>% dplyr::select(-c("Gene.Annotation"))))),
        column_order = c(),
        show_row_dend = F,
        show_column_dend = F,
        cluster_columns = T,
        col = col_fun,
        cluster_rows = T,
        top_annotation = samname,
        column_km = 2,
        row_km = 1,
        column_gap = unit(5, "mm"),
        row_gap = unit(5, "mm"),
        column_dend_height = unit(3, "cm"),
        row_dend_width = unit(3, "cm"),
        right_annotation = rowAnnotation(foo = anno_text(mat_bnt_order$Gene.Annotation)),
        show_row_names = F,
        heatmap_width = unit(25, "cm"),
        heatmap_height = unit(30, "cm"),
        show_column_names = F, 
        )
```

```{r Bingo results formatting for heatmaps SKYBLUE MODULE, include = F}
read.table(
  file = "/Users/benyoung/Dropbox/Projects/Acropora_palmata_disease_exposure/paper/GO_gene_lists/BINGO_results/skyblue_allGO_FDR_0.01.bgo",
  skip = 19,
  stringsAsFactors = F,
  sep = "\t",
  header = T
) %>%
  dplyr::select(GO.ID, Description, Genes.in.test.set) %>%
  separate_rows(Genes.in.test.set, sep = '\\|', convert = T) -> skyblue_results

wantGO <- c("GO:0008610")
#wantGO <- c("GO:0048584")

#Putting the GO and numbers before terms for imported bingo results
skyblue_results %>% 
  mutate(GO.ID_FULL = pad27(GO.ID)) %>%
  dplyr::filter(GO.ID_FULL %in% wantGO) ->
  intGOterms
#View(intGOterms)

forheatmap <- as.character(intGOterms$Genes.in.test.set)
#View(forheatmap)

colnames(skyblue_results)[colnames(skyblue_results)=="Genes.in.test.set"] <- "Gene.ID"
annot %>%
  dplyr::select(Gene.ID, Gene.Annotation) %>%
  inner_join(intGOterms %>% dplyr::rename(Gene.ID =Genes.in.test.set)) %>% View()

# Putting Gene.ID to upper case so can match to bingo results
#annot %>%
#  mutate(GENE.ID = toupper(Gene.ID)) %>%
#  dplyr::select(GENE.ID, GO_Terms_Swiss.Prot) %>%
#    tidyr::drop_na() %>% View()


```

```{r Skyblue Heatmap GO genes, fig.width=20, fig.height=20, echo = F}
mat_bnt <- assay(vsdyear)[forheatmap,]
#View(mat_bnt)
skyblue

tfall %>% 
  dplyr::select(Treatment) %>% 
  as.data.frame() -> bnt_tfall

ccann <- bnt_tfall
colnames(ccann) <- c("Treatment")
colcol <- list("Treatment"=c("baseline"="dodgerblue3", "exposed.no_transmission"="darkgoldenrod2", "exposed.transmission"="orangered3"))
samname <- HeatmapAnnotation(df = ccann,
                             which = "col",
                             col = colcol)

col_fun = colorRamp2(c(-2, 0, 2), c("dodgerblue4", "white", "red3"))
col_fun(seq(-3, 3))

t(scale(t(mat_bnt)))
annot %>% dplyr::select(Gene.ID, Gene.Annotation) %>% 
  right_join(mat_bnt %>% 
               as.data.frame() %>% 
               rownames_to_column(var="Gene.ID")) %>%
  column_to_rownames(var="Gene.ID") %>%
  tidyr::drop_na() -> matmatnorder -> mat_bnt_order

sample_order <- c()

Heatmap(t(scale(t(mat_bnt_order %>% as.data.frame() %>% dplyr::select(-c("Gene.Annotation"))))),
        column_order = c(),
        show_row_dend = F,
        show_column_dend = F,
        cluster_columns = T,
        col = col_fun,
        cluster_rows = T,
        top_annotation = samname,
        column_km = 2,
        row_km = 1,
        column_gap = unit(5, "mm"),
        row_gap = unit(5, "mm"),
        column_dend_height = unit(3, "cm"),
        row_dend_width = unit(3, "cm"),
        right_annotation = rowAnnotation(foo = anno_text(mat_bnt_order$Gene.Annotation)),
        show_row_names = F,
        heatmap_width = unit(25, "cm"),
        heatmap_height = unit(25, "cm"),
        show_column_names = F, 
        )
```

```{r Selecting genes in GO terms for BT genes, include=F}
read.table(
  file = "/Users/benyoung/Dropbox/Projects/Acropora_palmata_disease_exposure/paper/GO_gene_lists/BINGO_results/cyan_allGO_FDR_0.01.bgo",
  skip = 23,
  stringsAsFactors = F,
  sep = "\t",
  header = T
) %>%
  dplyr::select(GO.ID, Description, Genes.in.test.set) %>%
  separate_rows(Genes.in.test.set, sep = '\\|', convert = T) -> bt_results
#View(bt_results)

wantGO <- c("GO:0005539", "GO:0042151", "GO:0005764", "GO:0070062", "GO:0006955", "GO:0002252", "GO:0006956", "GO:0002526") #bioluminesence, response to endogenous stimulus, enzyme linked cell surace receptor signal00ling, reg of cytokine biosynthetic process

#Putting the GO and numbers before terms for imported bingo results
bt_results %>% 
  mutate(GO.ID_FULL = pad27(GO.ID)) %>%
  dplyr::filter(GO.ID_FULL %in% wantGO) ->
  intGOterms
#View(intGOterms)

forheatmap <- as.character(intGOterms$Genes.in.test.set)
forheatmap
#View(forheatmap)

annot %>%
  dplyr::select(Gene.ID, Gene.Annotation) %>%
  inner_join(intGOterms %>% dplyr::rename(Gene.ID = Genes.in.test.set)) -> PC1_GO_gene

write.csv(PC1_GO_gene, file = "~/Dropbox/Projects/Acropora_palmata_disease_exposure/paper/annotated_genelists/cyan_GO_annotated.csv")
# Putting Gene.ID to upper case so can match to bingo results
#annot %>%
#  mutate(GENE.ID = toupper(Gene.ID)) %>%
#  dplyr::select(GENE.ID, GO_Terms_Swiss.Prot) %>%
#    tidyr::drop_na() %>% View()
```

```{r BT Cell Adhesion GO Genes, fig.width=20, fig.height=20}
mat_bnt <- assay(vsdyear)[forheatmap,]
vsdyear

tfall %>% 
  dplyr::select(Treatment) %>% 
  as.data.frame() -> bnt_tfall

ccann <- bnt_tfall
colnames(ccann) <- c("Treatment")
colcol <- list("Treatment"=c("baseline"="dodgerblue3","exposed.no_transmission"="darkgoldenrod2", "exposed.transmission"="orangered3"))
samname <- HeatmapAnnotation(df = ccann,
                             which = "col",
                             col = colcol)


col_fun = colorRamp2(c(-2, 0, 2), c("dodgerblue4", "white", "red3"))
col_fun(seq(-3, 3))

# adding annotation into vsd count matrix
t(scale(t(mat_bnt)))
annot %>% dplyr::select(Gene.ID, Gene.Annotation) %>% 
  right_join(mat_bnt %>% 
               as.data.frame() %>% 
               rownames_to_column(var="Gene.ID")) %>%
  column_to_rownames(var="Gene.ID") %>%
  tidyr::drop_na() -> matmatnorder-> mat_bnt_order


# GO matrix (yes no) for genes matching to a term
PC1_GO_gene %>%
  dplyr::select(Gene.Annotation, Description, Gene.ID) %>%
  dplyr::mutate(membership=1) %>%
  tidyr::spread(Description, membership, fill=0) -> PC1_GO2gene_matrix

# merging by Gene.ID to make 1 matrix for heatmaps
mat_bnt_order %>%
  rownames_to_column(var="Gene.ID") %>%
  inner_join(PC1_GO2gene_matrix) %>%
  column_to_rownames(var="Gene.ID") -> mat_bnt_order


HM1 <- Heatmap(t(scale(t(mat_bnt_order %>% as.data.frame() %>% dplyr::select(-c("Gene.Annotation", 86:93))))),
        column_order = ,
        show_row_dend = F,
        show_column_dend = T,
        cluster_columns = T,
        col = col_fun,
        cluster_rows = T,
        top_annotation = samname,
        column_km = 2,
        row_km = 1,
        column_gap = unit(5, "mm"),
        row_gap = unit(5, "mm"),
        column_dend_height = unit(3, "cm"),
        row_dend_width = unit(3, "cm"),
        show_row_names = F,
        heatmap_width = unit(17, "cm"),
        heatmap_height = unit(50, "cm"),
        show_column_names = F)

HM2 <- Heatmap(t(t(mat_bnt_order %>% as.data.frame() %>% dplyr::select(c(86:93)))),
               show_row_dend = F,
               show_column_dend = F,
               heatmap_width = unit(15, "cm"),
               show_column_names = T,
               show_row_names = F,
               right_annotation = rowAnnotation(foo = anno_text(mat_bnt_order$Gene.Annotation)),
               col = c("white", "black"))

ht_list <- HM1 + HM2
draw(ht_list, ht_gap = unit(1, "cm"))
```