---
title: "Symbiodiniaceae_analysis"
author: "Ben Young"
date: "4/29/2019"
output: html_document
---

# Library and Function Loading

```{r package installation and loading, include=FALSE}
library(DEGreport)
library(Rgraphviz)
library(tximport)
library(tximportData)
library(readr)
library(DESeq2)
library(biomaRt)
library(pheatmap)
library(RColorBrewer)
library(vsn)
library(genefilter)
library(gplots)
library(ggplot2)
library(limma)
library(flashClust)
library(pathfindR)
library(edgeR)
library(vidger)
library(topGO)
library(ComplexHeatmap)
library(tidyr)
library(data.table)
library(ggplot2)
library(sciplot)
library(GOplot)
library(dplyr)
library(circlize)
library(tidyverse)
library(tibble)
```

```{r Function for Formatting Bingo Heatmaps, include = F}
pad27 <- function(x){
  str_x = as.character(x)
  digits = nchar(str_x)
  toAdd = 7-digits
  name = paste0('GO:',
                strrep(0,toAdd), 
                x)
  name
}
```

```{r Function for PC 2 and 3, include=FALSE}
#PCA 2 and 3 axis creation
pcaaxes23 = function (object, intgroup = "condition", ntop = 500, returnData = FALSE) 
{
    rv <- rowVars(assay(object))
    select <- order(rv, decreasing = TRUE)[seq_len(min(ntop, 
        length(rv)))]
    pca <- prcomp(t(assay(object)[select, ]))
    percentVar <- pca$sdev^2/sum(pca$sdev^2)
    if (!all(intgroup %in% names(colData(object)))) {
        stop("the argument 'intgroup' should specify columns of colData(dds)")
    }
    intgroup.df <- as.data.frame(colData(object)[, intgroup, 
        drop = FALSE])
    group <- if (length(intgroup) > 1) {
        factor(apply(intgroup.df, 1, paste, collapse = ":"))
    }
    else {
        colData(object)[[intgroup]]
    }
    d <- data.frame(PC2 = pca$x[, 2], PC3 = pca$x[, 3], group = group, 
        intgroup.df, name = colnames(object))
    if (returnData) {
        attr(d, "percentVar") <- percentVar[2:3]
        return(d)
    }
    ggplot(data = d, aes_string(x = "PC2", y = "PC3", color = "group")) + 
        geom_point(size = 3) + xlab(paste0("PC2: ", round(percentVar[2] * 
        100), "% variance")) + ylab(paste0("PC3: ", round(percentVar[3] * 
        100), "% variance")) + coord_fixed()
}
```

```{r Function for PC 3 and 4, include=FALSE}
#PCA 3 and 4 axis creation
pcaaxes34 = function (object, intgroup = "condition", ntop = 500, returnData = FALSE) 
{
    rv <- rowVars(assay(object))
    select <- order(rv, decreasing = TRUE)[seq_len(min(ntop, 
        length(rv)))]
    pca <- prcomp(t(assay(object)[select, ]))
    percentVar <- pca$sdev^2/sum(pca$sdev^2)
    if (!all(intgroup %in% names(colData(object)))) {
        stop("the argument 'intgroup' should specify columns of colData(dds)")
    }
    intgroup.df <- as.data.frame(colData(object)[, intgroup, 
        drop = FALSE])
    group <- if (length(intgroup) > 1) {
        factor(apply(intgroup.df, 1, paste, collapse = ":"))
    }
    else {
        colData(object)[[intgroup]]
    }
    d <- data.frame(PC3 = pca$x[, 3], PC4 = pca$x[, 4], group = group, 
        intgroup.df, name = colnames(object))
    if (returnData) {
        attr(d, "percentVar") <- percentVar[3:4]
        return(d)
    }
    ggplot(data = d, aes_string(x = "PC3", y = "PC4", color = "group")) + 
        geom_point(size = 3) + xlab(paste0("PC3: ", round(percentVar[3] * 
        100), "% variance")) + ylab(paste0("PC4: ", round(percentVar[4] * 
        100), "% variance")) + coord_fixed()
}

```

# Data Prep for Analysis

```{r Making Salmon Vectors Names, include=FALSE}
#Loading quantification data output from the slippery Salmon
PPall <- list.files(path = "~/Dropbox/PhD/Projects/apal_innate_immune_response/symbiont_analysis/symb_quant/", full.names = F, pattern = "\\_salmon")
FILESall <- file.path(PPall, "quant.sf")

names(FILESall) <- PPall
head(FILESall)

#all(file.exists("~/Dropbox/Projects/Acropora_palmata_disease_exposure/symbiont/symb/1_salmon/quant.sf"))
```

```{r Reading in Salmon Files, include=F}
#Importing the count data using salmon quant.sf files and the text to gene file
setwd("~/Dropbox/PhD/Projects/apal_innate_immune_response/symbiont_analysis/symb_quant//")
txi.salmon.countall <- tximport(files = FILESall, type = "salmon", txOut = TRUE )
str(txi.salmon.countall$counts)
#View(txi.salmon.countall$counts)
```

```{r Treatment File Import, include=FALSE}
tfall <- read.csv("~/Dropbox/PhD/Projects/apal_innate_immune_response/symbiont_analysis/symb_treatment_file.csv")
str(tfall)
tfall$Genotype = as.factor(tfall$Genotype)
tfall$Treatment_basic = as.factor(tfall$Treatment_basic)
tfall$Year = as.factor(tfall$Year)
str(tfall)
#View(tfall)
attach(tfall)
```

```{r Symb Annotation File, include = F}
kb8_annot.out <- read.delim("~/Dropbox/PhD/Projects/apal_innate_immune_response/symbiont_analysis/reference_transcriptome/kb8_annot.out.txt")
#View(kb8_annot.out)
kb8_annot.out %>% mutate(Gene.ID = toupper(Gene.ID)) -> kb8_annot.out
```

## Analysis
### Prefiltering

```{r Making DeSeq object, include=FALSE}
#making the dds model to use in deseq2
ddsall = DESeqDataSetFromTximport(txi.salmon.countall, tfall, ~ Treatment_basic)
```

```{r CPM filtering Adding to DeSeq Object and design change}
#checking rows (genes) and columns (samples) match
nrow(ddsall)
ncol(ddsall)

# cpm filtering step and seeing what original VS filtered gene number left is
cccall <- counts(ddsall)
keep <- rowSums(cpm(cccall)>=1) >= 20
cccall <- cccall[keep, ]

nrow(ddsall)
ncol(ddsall)
nrow(cccall)
ncol(cccall)

cccall %>%
  as.data.frame() %>%
  tibble::rownames_to_column(var="Gene.ID") %>%
  mutate(Gene.ID = toupper(Gene.ID)) %>%
  tibble::column_to_rownames(var = "Gene.ID") -> cccall

#write.table(cccall, file = "~/Desktop/DE/clust/all/genes4clust.txt", sep="\t", quote=F, row.names=T, col.names=T)

# re adding to the DeSeq2 object for analysis
ddsall <- DESeqDataSetFromMatrix(countData = cccall, colData = tfall, design = ~ Treatment_basic)

ddsall$group <- factor(paste0(ddsall$Treatment_basic, ddsall$Outcome))
design(ddsall) <- ~ Year + group
#(ddsall$group)
```

```{r}
cccall %>% as.data.frame() %>%
  tibble::rownames_to_column(var="Gene.ID") -> filtering

filtering <- c(filtering$Gene.ID)

kb8_annot.out %>%
  filter(Gene.ID %in% filtering) ->
  filtered_annot
# View(filtered_annot)
```

## Principal Component Analysis

```{r VST, include=FALSE}
## VSD stabilisation and visulisation
vsdall <- vst(ddsall, blind=FALSE)
#View(vsdall)
```

```{r VST with year removed, include = F}
vsdyear <- vst(ddsall)
assay(vsdyear) <- limma::removeBatchEffect(assay(vsdyear), vsdyear$Year)
```

```{r PCA Analysis}
cbPalette <- c("lightblue" ,"orange")
plotPCA(vsdall, intgroup=c("Treatment_basic"), returnData = F)
pca12 <- plotPCA(vsdall, intgroup=c("Treatment_basic"), returnData = TRUE)
ggplot(pca12, aes(PC1, PC2, color=Treatment_basic, shape=Year)) + 
  geom_point(size=3) +  xlab(paste0("PC1 82% variance")) + 
  ylab(paste0("PC2 7% variance")) + 
  theme(legend.position="right")  + 
  theme(text = element_text(size=24))  + 
  theme(legend.key.size = unit(2, "cm")) + 
  geom_point(size = 4) + 
  scale_colour_manual(values=cbPalette) +
  theme_bw()
#ggsave(filename = "pc12_treatbasic.jpeg", path = #"~/Dropbox/Projects/Acropora_palmata_disease_exposure/symbiont/figures/")

plotPCA(vsdall, intgroup=c("Genotype"), returnData = F)
pca12 <- plotPCA(vsdall, intgroup=c("Genotype"), returnData = TRUE)
ggplot(pca12, aes(PC1, PC2, color=Genotype, shape=Year)) + 
  geom_point(size=3) +  xlab(paste0("PC1 82% variance")) + 
  ylab(paste0("PC2 7% variance")) + 
  theme(legend.position="right")  + 
  theme(text = element_text(size=10))  + 
  theme(legend.key.size = unit(0.5, "cm")) + 
  geom_point(size = 4) + 
  theme_bw()
#ggsave(filename = "pc12_genotype.jpeg", path = #"~/Dropbox/Projects/Acropora_palmata_disease_exposure/symbiont/figures/")
```

```{r PCA Fig xx Symb PCA, echo = F}
cbPalette <- c("dodgerblue3" , "darkgoldenrod2", "orangered3")
plotPCA(vsdall,
        intgroup = c("Treatment"),
        returnData = F)
pca12 <- plotPCA(vsdall,
                 intgroup = c("Treatment"),
                 returnData = TRUE)

ggplot(pca12, aes(PC1, PC2, color = Treatment, shape = Year)) +
  geom_point(size = 3) +  xlab(paste0("PC1 82% variance")) +
  ylab(paste0("PC2 7% variance")) +
  theme(legend.position = "right")  +
  theme(
    text = element_text(size = 11, family = "Arial"),
    legend.position = "right",
    panel.background = element_rect(fill = "transparent"),
    axis.line = element_line(color = "black"),
    legend.key = element_rect(fill = "transparent"),
    axis.text = element_text(size = 8),
    legend.key.size = unit(0.7, "cm")
  )  +
  scale_colour_manual(values = cbPalette) +
  stat_ellipse(aes(PC1, PC2, group = Year), type = "norm") +
  scale_color_manual(
    name = "Treatments",
    values = c("dodgerblue3" , "darkgoldenrod2", "orangered3"),
    labels = c("Baseline", "No Transmission", "Transmission")
  )
#ggsave(filename = "pc12_treat.jpeg", path = #"~/Dropbox/Projects/Acropora_palmata_disease_exposure/symbiont/figures/")
```

```{r PCA Analysis 2 and 3, echo = F}
pcaaxes23(vsdall, intgroup=c("Treatment_basic"), returnData = F)
pca23 <- pcaaxes23(vsdall, intgroup=c("Treatment_basic"), returnData = TRUE)
ggplot(pca23, aes(PC2, PC3, colour=Treatment_basic)) + 
  geom_point(size=4) +  
  xlab(paste0("PC2 7% variance")) + 
  ylab(paste0("PC3 2% variance")) + 
  theme(legend.position="right")  + 
  theme(text = element_text(size=10))  + 
  theme(legend.key.size = unit(0.5, "cm")) + 
  geom_point(size = 4) +
  scale_colour_manual(values=cbPalette) +
  theme_bw()
#ggsave(filename = "pc23_treatbasic.jpeg", path = #"~/Dropbox/Projects/Acropora_palmata_disease_exposure/symbiont/figures/")

pcaaxes23(vsdall, intgroup=c("Genotype"), returnData = F)
pca23 <- pcaaxes23(vsdall, intgroup=c("Genotype"), returnData = TRUE)
ggplot(pca23, aes(PC2, PC3, colour=Genotype)) + 
  geom_point(size=4) +  
  xlab(paste0("PC2 7% variance")) + 
  ylab(paste0("PC3 2% variance")) + 
  theme(legend.position="right")  + 
  theme(text = element_text(size=10))  + 
  theme(legend.key.size = unit(0.5, "cm")) + 
  geom_point(size = 4) +
  theme_bw()
#ggsave(filename = "pc12_genotype.jpeg", path = #"~/Dropbox/Projects/Acropora_palmata_disease_exposure/symbiont/figures/")

pcaaxes23(vsdall,
          intgroup = c("Treatment"),
          returnData = F)
pca23 <-
  pcaaxes23(vsdall,
            intgroup = c("Treatment"),
            returnData = TRUE)
```

```{r Fig xx PCA23 symb, ewcho = F}
ggplot(pca23, aes(PC2, PC3, colour = Treatment, shape = Year)) +
  geom_point(size = 3) +
  xlab(paste0("PC2 7% variance")) +
  ylab(paste0("PC3 2% variance")) +
  theme(
    text = element_text(size = 11, family = "Arial"),
    legend.position = "right",
    panel.background = element_rect(fill = "transparent"),
    axis.line = element_line(color = "black"),
    legend.key = element_rect(fill = "transparent"),
    axis.text = element_text(size = 8),
    legend.key.size = unit(0.7, "cm")
  ) +
  scale_colour_manual(values = cbPalette) +
  stat_ellipse(pca23,
               mapping = aes(PC2, PC3, color = Treatment),
               inherit.aes = F) +
  scale_color_manual(
    name = "Treatments",
    values = c("dodgerblue3" , "darkgoldenrod2", "orangered3"),
    labels = c("Baseline", "No Transmission", "Transmission")
  )

stat_ellipse(pca23, mapping = aes(PC2, PC3, color = Treatment), inherit.aes = F)
#ggsave(filename = "pc23_treat.jpeg", path = #"~/Dropbox/Projects/Acropora_palmata_disease_exposure/symbiont/figures/")
```

```{r PCA34, echo = F}
pcaaxes34(vsdall, intgroup=c("Treatment"), returnData = F)
pca34 <- pcaaxes34(vsdall, intgroup=c("Treatment"), returnData = TRUE)
ggplot(pca34, aes(PC3, PC4, colour=Genotype, shape=Treatment)) + 
  geom_point(size=3) +  
  xlab(paste0("PC3 2% variance")) + 
  ylab(paste0("PC4 1% variance")) + 
  theme(legend.position="right")  + 
  theme(text = element_text(size=10))  + 
  theme(legend.key.size = unit(0.5, "cm")) + 
  geom_point(size = 3) +
  theme_bw() + 
  geom_polygon(fill=NA)
#ggsave(filename = "pc34_treat.jpeg", path = "~/Dropbox/Projects/Acropora_palmata_disease_exposure/symbiont/figures/")
```


## Analysis of Genes Driving PC1 Split 

```{r, Genes driving difference between PC1 and PC2} 
## getting the laodings for the PC's from PCA plots
DESeq2:::plotPCA.DESeqTransform
rv <- rowVars(assay(vsdall))
#View(rowvars)
ntop = 1000
select <- order(rv, decreasing = TRUE)[seq_len(min(ntop, 
        length(rv)))]
#View(select)

intgroup=c("Year", "Treatment_basic")
ntop = 1000
rv <- rowVars(assay(vsdall))
    select <- order(rv, decreasing = TRUE)[seq_len(min(ntop,
        length(rv)))]
    pca <- prcomp(t(assay(vsdall)[select, ]))
    percentVar <- pca$sdev^2/sum(pca$sdev^2)
    if (!all(intgroup %in% names(colData(vsdall)))) {
        stop("the argument 'intgroup' should specify columns of colData(dds)")
    }
    
rotations <- pca$rotation
rotations
#View(rotations)

pc1_rotations <- rotations[,c(1)]
#View(pc1_rotations)
hist(pc1_rotations, breaks = 30)

pc1_rotations <- as.data.frame(pc1_rotations)
str(pc1_rotations)
#View(pc1_rotations)

#interquartile range and selecting any genes outside of it to explain pc1 high variance explanation
quantile(pc1_rotations$pc1_rotations, 0.90)
quantile(pc1_rotations$pc1_rotations, 0.10)
quantile(pc1_rotations$pc1_rotations, 0.50)
median(pc1_rotations$pc1_rotations)

wanted_values <- subset(pc1_rotations, pc1_rotations >0.03632153  | pc1_rotations < 0.02566094  )
nrow(wanted_values)
View(wanted_values)

rownamespc1_variance <- row.names(wanted_values)
# View(rownamespc1_variance)
# str(rownamespc1_variance)
# str(hi)
#write.table(rownamespc1_variance,
#            file = "~/Dropbox/Projects/Acropora_palmata_disease_exposure/symbiont/PC1_analysis/wanted_genes#.txt",
#            sep = "\t", quote=F, row.names=F, col.names=F)

#rownamespc1_variance

#write.table(wanted_values, file = "~/Desktop/Final_analysis/PC1_variance/PC_loadings.txt", sep = "\t", quote=F, row.names=T, col.names=F)
```

```{r GO analysis prep for genes driving PC1, include = F}
#allrowsbnt
read.table(
  file = "/Users/benyoung/Dropbox/PhD/Projects/apal_innate_immune_response/host_analysis/GO_gene_lists/BINGO_results/PC1_symb_allGO)FDR_0.01.bgo",
  skip = 19,
  stringsAsFactors = F,
  sep = "\t",
  header = T
) %>%
  dplyr::select(GO.ID, Description, Genes.in.test.set) %>%
  separate_rows(Genes.in.test.set, sep = '\\|', convert = T) -> PC1_results

#(PC1_results)

wantGO <- c("GO:0006094", "GO:0006827", "GO:0042128", "GO:0015075")
#wantGO <- c("GO:0015075", "GO:0016021", "GO:0042301", "GO:0006094", "GO:0022904")
#wantGO <- c("GO:0006827", "GO:0006094", "GO:0042128", "GO:0018298", "GO:0022904")

#Putting the GO and numbers before terms for imported bingo results
PC1_results %>% 
  mutate(GO.ID_FULL = pad27(GO.ID)) %>%
  dplyr::filter(GO.ID_FULL %in% wantGO) ->
  intGOterms

View(intGOterms)

forheatmap <- as.character(intGOterms$Genes.in.test.set)

#View(PC1_GO_gene)
```

```{r Gene Lists PC1 Supplementary}
kb8_annot.out %>% dplyr::select(Gene.ID, Gene.annotation, Swisprot) %>%
  inner_join(wanted_values %>% 
               as.data.frame() %>% 
               rownames_to_column(var="Gene.ID")) -> PC1_annotated_supp
#View(PC1_annotated_supp)

#write.csv(PC1_annotated_supp, 
#          file = "~/Desktop/PC1_loadings.csv",
#          row.names = F)

read.table(
  file = "/Users/benyoung/Dropbox/PhD/Projects/apal_innate_immune_response/host_analysis//GO_gene_lists/BINGO_results/PC1_symb_allGO)FDR_0.01.bgo",
  skip = 19,
  stringsAsFactors = F,
  sep = "\t",
  header = T
) %>% mutate(GO.ID_FULL = pad27(GO.ID)) %>%
  dplyr::select(-c(1)) %>%
  dplyr::select(c(9,7,1,2,3,4,8)) -> PC1_symb_bingo_results

#write.csv(PC1_symb_bingo_results, 
#          file = "~/Desktop/PC1_symb_bingo_results.csv",
#          row.names = F)
```

```{r Figxx PC1 variance SYMB complexheatmap, fig.height=50, fig.width=40}
matmatnorder <-(assay(vsdall)[rownames(assay(vsdall)) %in% forheatmap,])

# Colours for the years
ccann <- data.frame(tfall$Year)
colnames(ccann) <- c("Year")
colcol <- list(
  "Year" =
    c(
      "2016" = "grey", 
      "2017" = "black"))

# Top annotation bar
samname <- HeatmapAnnotation(df = ccann,
                             which = "col",
                             col = colcol,
                             simple_anno_size = unit(1.5, "cm"),
                             annotation_name_gp = gpar(fontsize = 70),
                             annotation_name_side = "left")

# Fill for the gene VST data
col_fun = colorRamp2(c(-2, 0, 2), c("dodgerblue4", "white", "red3"))
col_fun(seq(-3, 3))

# adding annotation into vsd count matrix
filtered_annot %>% dplyr::select(Gene.ID, Gene.annotation) %>% 
  right_join(matmatnorder %>% 
               as.data.frame() %>% 
               rownames_to_column(var="Gene.ID")) %>%
  column_to_rownames(var="Gene.ID") %>%
  tidyr::drop_na() -> matmatnorder

# GO matrix (yes no) for genes matching to a term
PC1_GO_gene %>%
  dplyr::select(Gene.annotation, Description, Gene.ID) %>%
  dplyr::mutate(membership=1) %>%
  tidyr::spread(Description, membership, fill=0) -> PC1_GO2gene_matrix

# merging by Gene.ID to make 1 matrix for heatmaps
matmatnorder %>%
  rownames_to_column(var="Gene.ID") %>%
  inner_join(PC1_GO2gene_matrix) %>%
  column_to_rownames(var="Gene.ID") -> matmatnorder

# Changing names to a character
matmatnorder$Gene.annotation <- as.character(matmatnorder$Gene.annotation)
#View(matmatnorder)

#matmatnorder %>%
#  rownames_to_column(var="genes") %>%
#  inner_join(GO_long_PC1, by = "genes") %>%
#  column_to_rownames(var="genes") %>% head()

#id_to_annot #need to have rownames when putting into complex heatmap
#hello <- rowAnnotation(foo = anno_text(matmatnorder %>% as.data.frame() %>% dplyr::select(Gene.annotation)))

HM1 <- Heatmap(t(scale(t(matmatnorder %>% as.data.frame() %>% dplyr::select(-c("Gene.annotation", 85:88))))),
        cluster_columns = T,
        show_row_dend = F,
        show_column_dend = T,
        column_km = 2,
        row_km = 2,
        col = col_fun,
        top_annotation = samname,
        column_gap = unit(2, "cm"),
        row_gap = unit(2, "cm"),
        column_dend_height = unit(20, "cm"),
        row_dend_width = unit(8, "cm"),
        show_row_names = F,
        heatmap_width = unit(70, "cm"),
        heatmap_height = unit(200, "cm"),
        row_title_gp = gpar(fontsize = 80, font = 1),
        column_title_gp = gpar(fontsize = 80, font = 1),
        show_column_names = F,
        row_title = NULL,
        column_title = NULL)

HM2 <- Heatmap(t(t(matmatnorder %>% as.data.frame() %>% dplyr::select(c(85:88)))),
               show_row_dend = F,
               show_column_dend = F,
               heatmap_width = unit(105, "cm"),
               show_column_names = T,
               show_row_names = F, 
               column_names_rot = 30,
               right_annotation = rowAnnotation(foo = anno_text(matmatnorder$Gene.annotation, gp=gpar(fontsize=70))),
               column_names_gp = gpar(fontsize = 50, font = 1),
               col = c("white", "black"))

ht_list <- HM1 + HM2
draw(ht_list, ht_gap = unit(1, "cm"))
```

```{r}
Annotation_file_for_analysis <- read.delim("~/Dropbox/PhD/Projects/apal_innate_immune_response/symbiont_analysis/reference_transcriptome/kb8_annot.out.txt")
annot <- Annotation_file_for_analysis
View(annot)
str(annot)
```

### Photosynthesis Heatmap

```{r CAF for cytoscape}
#allrowsbnt
read.table(
  file = "/Users/benyoung/Dropbox/PhD/Projects/apal_innate_immune_response/host_analysis/GO_gene_lists/BINGO_results/PC1_symb_allGO)FDR_0.01.bgo",
  skip = 19,
  stringsAsFactors = F,
  sep = "\t",
  header = T
) %>%
  dplyr::select(GO.ID, Description, Genes.in.test.set) %>%
  separate_rows(Genes.in.test.set, sep = '\\|', convert = T) -> PC1_results

#View(PC1_results)

wantGO <- c("GO:0019684", "GO:0022900", "GO:0042651", "GO:0009522", "GO:0009523", "GO:0030077", "GO:0009535")
#wantGO <- c("GO:0015075", "GO:0016021", "GO:0042301", "GO:0006094", "GO:0022904")
#wantGO <- c("GO:0006827", "GO:0006094", "GO:0042128", "GO:0018298", "GO:0022904")

#Putting the GO and numbers before terms for imported bingo results
PC1_results %>% 
  mutate(GO.ID_FULL = pad27(GO.ID)) %>%
  dplyr::filter(GO.ID_FULL %in% wantGO) ->
  intGOterms
View(intGOterms)

forheatmap <- as.character(intGOterms$Genes.in.test.set)

filtered_annot %>%
  dplyr::select(Gene.ID, Gene.annotation) %>%
  inner_join(intGOterms %>% dplyr::rename(Gene.ID = Genes.in.test.set)) -> PC1_GO_gene

View(PC1_GO_gene)
```

```{r PC1 variance complexheatmap, fig.height=100, fig.width=70}
matmatnorder <- assay(vsdyear)[forheatmap,]
#View(assay(vsdall))

# Colours for the years
ccann <- data.frame(tfall$Year)
colnames(ccann) <- c("Year")
colcol <- list(
  "Year" =
    c(
      "2016" = "grey", 
      "2017" = "black"))

# Top annotation bar
samname <- HeatmapAnnotation(df = ccann,
                             which = "col",
                             col = colcol,
                             simple_anno_size = unit(1.5, "cm"),
                             annotation_name_gp = gpar(fontsize = 70),
                             annotation_name_side = "left")

# Fill for the gene VST data
col_fun = colorRamp2(c(-2, 0, 2), c("dodgerblue4", "white", "red3"))
col_fun(seq(-3, 3))

# adding annotation into vsd count matrix
filtered_annot %>% dplyr::select(Gene.ID, Gene.annotation) %>% 
  right_join(matmatnorder %>% 
               as.data.frame() %>% 
               rownames_to_column(var="Gene.ID")) %>%
  column_to_rownames(var="Gene.ID") %>%
  tidyr::drop_na() -> matmatnorder

# GO matrix (yes no) for genes matching to a term
PC1_GO_gene %>%
  dplyr::select(Gene.annotation, Description, Gene.ID) %>%
  dplyr::mutate(membership=1) %>%
  tidyr::spread(Description, membership, fill=0) -> PC1_GO2gene_matrix

# merging by Gene.ID to make 1 matrix for heatmaps
matmatnorder %>%
  rownames_to_column(var="Gene.ID") %>%
  inner_join(PC1_GO2gene_matrix) %>%
  column_to_rownames(var="Gene.ID") -> matmatnorder

# Changing names to a character
matmatnorder$Gene.annotation <- as.character(matmatnorder$Gene.annotation)
#View(matmatnorder)

#matmatnorder %>%
#  rownames_to_column(var="genes") %>%
#  inner_join(GO_long_PC1, by = "genes") %>%
#  column_to_rownames(var="genes") %>% head()

#id_to_annot #need to have rownames when putting into complex heatmap
#hello <- rowAnnotation(foo = anno_text(matmatnorder %>% as.data.frame() %>% dplyr::select(Gene.annotation)))
View(matmatnorder)

HM1 <- Heatmap(t(scale(t(matmatnorder %>% as.data.frame() %>% dplyr::select(-c("Gene.annotation", 86:92))))),
        cluster_columns = T,
        show_row_dend = F,
        show_column_dend = T,
        column_km = 0,
        row_km = 2,
        col = col_fun,
        top_annotation = samname,
        column_gap = unit(2, "cm"),
        row_gap = unit(2, "cm"),
        column_dend_height = unit(8, "cm"),
        row_dend_width = unit(8, "cm"),
        show_row_names = F,
        heatmap_width = unit(60, "cm"),
        heatmap_height = unit(210, "cm"),
        row_title_gp = gpar(fontsize = 70),
        column_title_gp = gpar(fontsize = 70),
        show_column_names = F)

HM2 <- Heatmap(t(t(matmatnorder %>% as.data.frame() %>% dplyr::select(c(86:92)))),
               show_row_dend = F,
               show_column_dend = F,
               heatmap_width = unit(85, "cm"),
               show_column_names = T,
               show_row_names = F, 
               column_names_rot = 30,
               right_annotation = rowAnnotation(foo = anno_text(matmatnorder$Gene.annotation, gp=gpar(fontsize=70))),
               column_names_gp = gpar(fontsize = 58),
               col = c("white", "black"))

ht_list <- HM1 + HM2
draw(ht_list, ht_gap = unit(1, "cm"))
```
